
<!DOCTYPE html>


<html lang="en" data-content_root="../../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>6.1 Exploratory Data Analysis - Protein &#8212; Data Chemistry 2025</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../../../_static/oldcustom.css?v=e52fa269" />
    <link rel="stylesheet" type="text/css" href="../../../_static/custom.css?v=ba2d5b83" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'content/modules/06-BioAssay/06-2-eda-protein';</script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="Exploratory Data Analysis 2:Chemical Modulation of Renin-Angiotensin-Aldosterone system (RAAS)" href="06-3-eda-pathway.html" />
    <link rel="prev" title="6.1 Bioactivity Primer" href="06-1-Bioactivity_Primer.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../../_static/logo.png" class="logo__image only-light" alt="Data Chemistry 2025 - Home"/>
    <script>document.write(`<img src="../../../_static/logo.png" class="logo__image only-dark" alt="Data Chemistry 2025 - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../index.html">
                    2025 Data Chemistry OLCC
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../01-OLCC-Primer/README.html">Module 1: OLCC Primer</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../01-OLCC-Primer/01-1-OLCC-Primer.html">Module 1.1: OLCC Primer</a></li>







<li class="toctree-l2"><a class="reference internal" href="../01-OLCC-Primer/01-2-python-basics.html">1.2 Python Basics</a></li>


<li class="toctree-l2"><a class="reference internal" href="../01-OLCC-Primer/01-3-python-intermediate.html">1.3 Intermediate Python</a></li>


</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../02-Introduction/README.html">Module 2: Introduction</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../02-Introduction/02-1-PubChemDataTypes.html">2.1 PubChem Data Types</a></li>



<li class="toctree-l2"><a class="reference internal" href="../02-Introduction/02-2-PUG-REST-activity.html">2.2 PUG-RESTY Activity</a></li>


</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../03-Public-Chemical-Databases/README.html">Module 3: Public Chemical Databases</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../03-Public-Chemical-Databases/03-01-Survey-Public-Chemical-Databases.html">Module 1.1: Public Compound Databases</a></li>



</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../04-Chemical-Representations/README.html">Module 4: Chemical Representations</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../04-Chemical-Representations/04-01-introduction_chemical_representations.html">4.1 Chemical Representation and Introduction to RDKit</a></li>







<li class="toctree-l2"><a class="reference internal" href="../04-Chemical-Representations/04-02-PubChem3D_update.html">4.2 PubChem 3D</a></li>


<li class="toctree-l2"><a class="reference internal" href="../04-Chemical-Representations/04-03-Substructure_Searching_with_SMARTS.html">4.3 SMARTS Substructure Searching</a></li>


</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../05-Searching-Chemical-Databases/README.html">Module 5: Searching Chemical Databases</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../05-Searching-Chemical-Databases/05-01-Exploring-PubChem.html">5.1 Exploring PubChem</a></li>



<li class="toctree-l2"><a class="reference internal" href="../05-Searching-Chemical-Databases/05-02-structure-inputs.html">5.2 Searching with Structures</a></li>



<li class="toctree-l2"><a class="reference internal" href="../05-Searching-Chemical-Databases/05-03-list-conversion.html">5.3 List Conversions</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="README.html">Module 6: BioAssays</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="06-1-Bioactivity_Primer.html">6.1 Bioactivity Primer</a></li>





<li class="toctree-l2 current active"><a class="current reference internal" href="#">6.1 Exploratory Data Analysis - Protein</a></li>



<li class="toctree-l2"><a class="reference internal" href="06-3-eda-pathway.html">Exploratory Data Analysis 2:<br/><br/>Chemical Modulation of Renin-Angiotensin-Aldosterone system (RAAS)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../07-QSAR-QSPR/README.html">Module 7: Quantitative Structure Property Reationships</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../07-QSAR-QSPR/07-01-QSPR.html">Molecular Descriptors</a></li>








</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../08-Molecular-Similarity/README.html">Module 8: Molecular Similarity</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../08-Molecular-Similarity/08-1-MolSimularity.html">Fingerprint Generation</a></li>



<li class="toctree-l2"><a class="reference internal" href="../08-Molecular-Similarity/08-2-MolSimularity.html">What is a bit vector?</a></li>


<li class="toctree-l2"><a class="reference internal" href="../08-Molecular-Similarity/08-3-MolSimularity.html">Objectives</a></li>





</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../09-CADDD/README.html">Module 9: Computer-Aided Drug Discovery and Design</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../09-CADDD/09-1-virtual-screening.html">Virtual Screening</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../10-Machine-Learning/README.html">Module 10: Supervised ML</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../10-Machine-Learning/10-01-supervised-ML.html">Machine Learning Basics</a></li>





</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../appendices/README.html">Appendices</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../appendices/Pandas.html">Introduction to Pandas</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../appendices/MBP1_CellStuctures.html">Molecular Biology Primer for Cheminformatics</a></li>

<li class="toctree-l2"><a class="reference internal" href="../../appendices/MBP2_Genes.html">Introduction to Genes and DNA</a></li>







</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-launch-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Launch interactive content">
    <i class="fas fa-rocket"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://mybinder.org/v2/gh/DivCHED-CCCE/DataChemistry2025OLCC/master?urlpath=lab/tree/docs/content/modules/06-BioAssay/06-2-eda-protein.ipynb" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch on Binder"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img alt="Binder logo" src="../../../_static/images/logo_binder.svg">
  </span>
<span class="btn__text-container">Binder</span>
</a>
</li>
      
      
      
      
      <li><a href="https://nanohub.org/tools/datachemolcc/hub/user-redirect/git-pull?repo=https%3A//github.com/DivCHED-CCCE/DataChemistry2025OLCC&urlpath=lab/tree/DataChemistry2025OLCC/docs/content/modules/06-BioAssay/06-2-eda-protein.ipynb&branch=master" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch on JupyterHub"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img alt="JupyterHub logo" src="../../../_static/images/logo_jupyterhub.svg">
  </span>
<span class="btn__text-container">JupyterHub</span>
</a>
</li>
      
      
      
      
      <li><a href="https://colab.research.google.com/github/DivCHED-CCCE/DataChemistry2025OLCC/blob/master/docs/content/modules/06-BioAssay/06-2-eda-protein.ipynb" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch on Colab"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img alt="Colab logo" src="../../../_static/images/logo_colab.png">
  </span>
<span class="btn__text-container">Colab</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/DivCHED-CCCE/DataChemistry2025OLCC" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/DivCHED-CCCE/DataChemistry2025OLCC/issues/new?title=Issue%20on%20page%20%2Fcontent/modules/06-BioAssay/06-2-eda-protein.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../../_sources/content/modules/06-BioAssay/06-2-eda-protein.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>6.1 Exploratory Data Analysis - Protein</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">6.1 Exploratory Data Analysis - Protein</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#exploratory-data-analysis-1-concordance-of-thyroid-hormone-receptor-activity-in-rats-and-humans">Exploratory Data Analysis 1: <br/><br/> Concordance of Thyroid Hormone Receptor Activity β in Rats and Humans</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#checking-the-amount-of-available-data">Checking the amount of available data</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#loading-data-for-rat-thyroid-hormone-receptor">Loading data for rat thyroid hormone receptor</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#checking-the-type-and-amount-of-available-data-for-each-species">Checking the type and amount of available data for each species</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#reviewing-the-available-bioassays">Reviewing the available bioassays</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#comparing-hts-screening-data-for-rat-and-human">Comparing HTS screening data for Rat and human</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#loading-the-hts-screening-data-for-rats">Loading the HTS screening data for Rats</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#substance-level-analysis">Substance-level analysis</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#activity-outcomes-and-phenotypes">Activity outcomes and phenotypes</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#efficacy">Efficacy</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#potency">Potency</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#efficacy-vs-potency">Efficacy vs Potency</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#compound-level-analysis">Compound-level Analysis</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#aggregating-substance-data-rats">Aggregating substance data rats</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#aggregating-substance-data-for-humans">Aggregating substance data for humans</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#merging-rat-data-and-human-data">Merging rat data and human data</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#compound-level-comparison">Compound-level comparison</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="exploratory-data-analysis-protein">
<span id="modules-06-bioassay-06-2-eda-protein"></span><h1>6.1 Exploratory Data Analysis - Protein<a class="headerlink" href="#exploratory-data-analysis-protein" title="Link to this heading">#</a></h1>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="exploratory-data-analysis-1-concordance-of-thyroid-hormone-receptor-activity-in-rats-and-humans">
<h1>Exploratory Data Analysis 1: <br><br> Concordance of Thyroid Hormone Receptor Activity β in Rats and Humans<a class="headerlink" href="#exploratory-data-analysis-1-concordance-of-thyroid-hormone-receptor-activity-in-rats-and-humans" title="Link to this heading">#</a></h1>
<p>In the preclinical stage of drug development, drug candidates are tested in non-human species.  One of the commonly used animals in the preclinical stage is rodents (e.g., mice and rats).</p>
<p>The underlying assumption here is that the bioactivity (e.g., toxicity and efficacy) of a drug candidate in animals is similar to its activity in humans.</p>
<p>In this notebook, we will look into the validity of this assumption by comparing the bioactivity data available in PubChem for human and rat thyroid hormone receptor β.</p>
<p>Recall we looked at two PubChem protein summary pages for accessions <code class="docutils literal notranslate"><span class="pre">NP_001257783</span></code> and <code class="docutils literal notranslate"><span class="pre">P10828</span></code> which were for the Norway Rat and Human thyroid hormone receptor β proteins.</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">NP_001257783</span></code> https://pubchem.ncbi.nlm.nih.gov/protein/NP_001257783  (rat)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">P10828</span></code> https://pubchem.ncbi.nlm.nih.gov/protein/P10828  (Human)</p></li>
</ul>
<p>Let’s import necessary libraries.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">seaborn</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">sns</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">requests</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">scipy.stats</span><span class="w"> </span><span class="kn">import</span> <span class="n">linregress</span>
<span class="c1">#import time</span>
<span class="c1">#import json</span>
</pre></div>
</div>
</div>
</div>
<section id="checking-the-amount-of-available-data">
<h2>Checking the amount of available data<a class="headerlink" href="#checking-the-amount-of-available-data" title="Link to this heading">#</a></h2>
<p>This section focuses on assessing the availability of the bioactivity data for the thyroid hormone receptor in both human and rat models.  The data sets were downloaded in August 2025 from the PubChem Protein summary pages for protein assesions  <code class="docutils literal notranslate"><span class="pre">NP_001257783</span></code> (rats) and <code class="docutils literal notranslate"><span class="pre">P10828</span></code> (humans).</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">NP_001257783</span></code> (rats)<br />
https://pubchem.ncbi.nlm.nih.gov/protein/NP_001257783#section=Chemicals-and-Bioactivities</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">P10828</span></code> (humans)<br />
https://pubchem.ncbi.nlm.nih.gov/protein/P10828#section=Chemicals-and-Bioactivities</p></li>
</ul>
<p>The names of the downloaded files are:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">file_protein_rat</span> <span class="o">=</span> <span class="s2">&quot;pubchem_protacxn_NP_001257783_bioactivity_protein.zip&quot;</span>
<span class="n">file_protein_human</span> <span class="o">=</span> <span class="s2">&quot;pubchem_protacxn_P10828_bioactivity_protein.zip&quot;</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-danger">
Note that the csv files are very large so we have provided the data in a compressed format. We will use Pandas Dataframes for our exploratory analysis, which can directly open a zip file, so there is no need to unzip them after you download them. 
</div><section id="loading-data-for-rat-thyroid-hormone-receptor">
<h3>Loading data for rat thyroid hormone receptor<a class="headerlink" href="#loading-data-for-rat-thyroid-hormone-receptor" title="Link to this heading">#</a></h3>
<p>The following cells load and provide an initial exploration of the bioactivity data for the <strong>rat thyroid hormone receptor</strong>.</p>
<p>This cell defines the data type for the “COMPOUND_CID” column to ensure accurate data loading and to prevent memory-related issues.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Define CID data type for reading CSV files to optimize memory usage</span>
<span class="n">mydtype</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;Compound_CID&quot;</span> <span class="p">:</span> <span class="s2">&quot;Int64&quot;</span><span class="p">}</span>
<span class="c1">#mydtype = {&quot;Compound_CID&quot; : &quot;Int64&quot;, &quot;pmid&quot;:&quot;Int64&quot;, &quot;taxid&quot;:&quot;object&quot;, &quot;targettaxid&quot;:&quot;object&quot;}</span>
</pre></div>
</div>
</div>
</div>
<p>We can now read our zip files containing bioactivity data in csv format for the rat thyroid hormone receptor into a Pandas DataFrame and then display the dimensions of the DataFrame. Notice that the read_csv method allows for direct reading of compressed files directly by using the <code class="docutils literal notranslate"><span class="pre">compression</span> <span class="pre">=</span> <span class="pre">'zip'</span></code> argument.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_rat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;pubchem_protacxn_NP_001257783_bioactivity_protein.zip&#39;</span><span class="p">,</span> <span class="n">compression</span><span class="o">=</span><span class="s1">&#39;zip&#39;</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="n">mydtype</span><span class="p">,</span><span class="n">low_memory</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">df_rat</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(31458, 32)
</pre></div>
</div>
</div>
</div>
<p>Display a first few rows of the rat bioactivity DataFrame to get a quick overview of the data’s structure and content.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_rat</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Bioactivity_ID</th>
      <th>Activity</th>
      <th>BioAssay_AID</th>
      <th>Substance_SID</th>
      <th>Compound_CID</th>
      <th>refsid</th>
      <th>Gene_ID</th>
      <th>PMID</th>
      <th>Aid_Type</th>
      <th>Last_Modified_Date</th>
      <th>...</th>
      <th>Representative_Protein_Accession</th>
      <th>Taxonomy_ID</th>
      <th>Cell_ID</th>
      <th>Target_Taxonomy_ID</th>
      <th>Anatomy_ID</th>
      <th>Anatomy</th>
      <th>dois</th>
      <th>pmcids</th>
      <th>pclids</th>
      <th>citations</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>394010671</td>
      <td>Active</td>
      <td>743066</td>
      <td>144203580</td>
      <td>5920</td>
      <td>NaN</td>
      <td>24831</td>
      <td>NaN</td>
      <td>Confirmatory</td>
      <td>20140820</td>
      <td>...</td>
      <td>NP_001257783</td>
      <td>10116</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>394011094</td>
      <td>Active</td>
      <td>743066</td>
      <td>144204055</td>
      <td>92044441</td>
      <td>NaN</td>
      <td>24831</td>
      <td>NaN</td>
      <td>Confirmatory</td>
      <td>20140820</td>
      <td>...</td>
      <td>NP_001257783</td>
      <td>10116</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>394011748</td>
      <td>Active</td>
      <td>743066</td>
      <td>144203868</td>
      <td>5803</td>
      <td>NaN</td>
      <td>24831</td>
      <td>NaN</td>
      <td>Confirmatory</td>
      <td>20140820</td>
      <td>...</td>
      <td>NP_001257783</td>
      <td>10116</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>394012257</td>
      <td>Active</td>
      <td>743066</td>
      <td>144204363</td>
      <td>444795</td>
      <td>NaN</td>
      <td>24831</td>
      <td>NaN</td>
      <td>Confirmatory</td>
      <td>20140820</td>
      <td>...</td>
      <td>NP_001257783</td>
      <td>10116</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>4</th>
      <td>394012468</td>
      <td>Active</td>
      <td>743066</td>
      <td>144204799</td>
      <td>57333099</td>
      <td>NaN</td>
      <td>24831</td>
      <td>NaN</td>
      <td>Confirmatory</td>
      <td>20140820</td>
      <td>...</td>
      <td>NP_001257783</td>
      <td>10116</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 32 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_rat</span><span class="o">.</span><span class="n">columns</span>    <span class="c1"># Uncomment to check the column names.</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Index([&#39;Bioactivity_ID&#39;, &#39;Activity&#39;, &#39;BioAssay_AID&#39;, &#39;Substance_SID&#39;,
       &#39;Compound_CID&#39;, &#39;refsid&#39;, &#39;Gene_ID&#39;, &#39;PMID&#39;, &#39;Aid_Type&#39;,
       &#39;Last_Modified_Date&#39;, &#39;Has_Dose_Response_Curve&#39;, &#39;RNAi_BioAssay&#39;,
       &#39;Protein_Accession&#39;, &#39;Activity_Type&#39;, &#39;Activity_Qualifier&#39;,
       &#39;Activity_Value&#39;, &#39;Bioassay_Data_Source&#39;, &#39;BioAssay_Name&#39;,
       &#39;Compound_Name&#39;, &#39;Target_Name&#39;, &#39;Target_Link&#39;, &#39;ECs&#39;,
       &#39;Representative_Protein_Accession&#39;, &#39;Taxonomy_ID&#39;, &#39;Cell_ID&#39;,
       &#39;Target_Taxonomy_ID&#39;, &#39;Anatomy_ID&#39;, &#39;Anatomy&#39;, &#39;dois&#39;, &#39;pmcids&#39;,
       &#39;pclids&#39;, &#39;citations&#39;],
      dtype=&#39;object&#39;)
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<p><b>Loading data for human thyroid hormone receptor</b></p>
<p>In the next two code cells:</p>
<ol class="arabic simple">
<li><p>Read the CSV file containing bioactivity data for the <strong>human</strong> thyroid hormone receptor into a pandas DataFrame, called <code class="docutils literal notranslate"><span class="pre">df_human</span></code> and then display the dimensions of the DataFrame.</p></li>
<li><p>Display the first 5 rows of the dataframe.</p></li>
</ol>
</div><div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your code here to load human data into the datafram and display dimensions</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your code here to display the first 5 rows of the human dataframe</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#df_human.columns    # Uncomment to check the column names.</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="checking-the-type-and-amount-of-available-data-for-each-species">
<h3>Checking the type and amount of available data for each species<a class="headerlink" href="#checking-the-type-and-amount-of-available-data-for-each-species" title="Link to this heading">#</a></h3>
<p>Here, we compare the types of bioactivity data available for both rat and human thyroid hormone receptors.</p>
<p>This cell counts the occurrences of each unique value in the <code class="docutils literal notranslate"><span class="pre">'Activity_Type'</span></code> column of the rat bioactivity DataFrame.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_rat</span><span class="p">[</span><span class="s1">&#39;Activity_Type&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Activity_Type
Potency    31458
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<ol class="arabic simple">
<li><p>Get the occurrences of each unique value in the <code class="docutils literal notranslate"><span class="pre">'Activity_Type'</span></code> column of the human bioactivity DataFrame in the next code cell.</p></li>
<li><p>Comment on the Activity types found in the rat vs. human data.</p></li>
</ol>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code here</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-success"> 
<strong>Python Dive:</strong> In the last few lines of code you used Pandas DataFrame methods to explore the data. Let’s take a closer look.
<div class="alert alert-block alert-info">
<details>
    <summary>Explanation of the code</summary>
<p>A <strong>method</strong> is a function bound to an object and uses the <code class="docutils literal notranslate"><span class="pre">.method</span></code> notation to attach it to the object.<br />
Here the object is a Pandas DataFrame called <code class="docutils literal notranslate"><span class="pre">df_human</span></code>.</p>
<ul class="simple">
<li><p>Some methods <strong>perform actions</strong> on the DataFrame and require parentheses with optional arguments, e.g. <code class="docutils literal notranslate"><span class="pre">df_human.head(10)</span></code> returns the first 10 rows.</p></li>
<li><p>Some properties are <strong>attributes</strong> rather than methods, so you access them without parentheses, e.g. <code class="docutils literal notranslate"><span class="pre">df_human.shape</span></code> returns the dimensions of the DataFrame.</p></li>
<li><p><strong><code class="docutils literal notranslate"><span class="pre">df_human.shape</span></code></strong> → Returns a tuple <code class="docutils literal notranslate"><span class="pre">(rows,</span> <span class="pre">columns)</span></code> with dataset size.<br />
<em>Example: <code class="docutils literal notranslate"><span class="pre">(5000,</span> <span class="pre">20)</span></code> means 5000 rows × 20 columns.</em></p></li>
<li><p><strong><code class="docutils literal notranslate"><span class="pre">df_human.head()</span></code></strong> → Displays the first 5 rows (or more with <code class="docutils literal notranslate"><span class="pre">head(n)</span></code>).<br />
<em>Great for previewing structure and sample values.</em></p></li>
<li><p><strong><code class="docutils literal notranslate"><span class="pre">df_human.columns</span></code></strong> → Lists the column names as an Index object.<br />
<em>Helps you see available variables for analysis.</em></p></li>
<li><p><strong><code class="docutils literal notranslate"><span class="pre">df_human['Activity_Type'].value_counts(dropna=False)</span></code></strong> → Counts frequency of each unique value.<br />
<em><code class="docutils literal notranslate"><span class="pre">dropna=False</span></code> ensures missing values (<code class="docutils literal notranslate"><span class="pre">NaN</span></code>) are included in the count.</em></p></li>
</ul>
</details>
</div>
</div>
<p>Sometimes, visualizations can make patterns in the data easier to see. Let’s create bar charts to compare the distribution of activity types in the rat and human bioactivity datasets. The following code uses the Python <strong><a class="reference external" href="https://seaborn.pydata.org/">Seaborn statistical data visualization package</a></strong> to create two side-by-side bar charts, one for the rat data and one for the human data. The <code class="docutils literal notranslate"><span class="pre">sharey=True</span></code> argument ensures both charts use the same y-axis scale for easier comparison.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">countplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_rat</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Activity_Type&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sns</span><span class="o">.</span><span class="n">countplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_human</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Activity_Type&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Number of Data Points (rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Number of Data Points (human)&quot;</span><span class="p">)</span>

<span class="k">for</span> <span class="n">ax</span> <span class="ow">in</span> <span class="n">axes</span> <span class="p">:</span>    <span class="c1"># loop over each chart.</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_xscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">container</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">containers</span><span class="p">:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">bar_label</span><span class="p">(</span><span class="n">container</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<ol class="arabic simple">
<li><p>What do you notice about the activity types available in the rat dataset compared to the human dataset?</p></li>
<li><p>If we want to compare small-molecule bioactivity between rats and humans, which activity type should we use, and why?</p></li>
</ol>
</div></section>
<section id="reviewing-the-available-bioassays">
<h3>Reviewing the available bioassays<a class="headerlink" href="#reviewing-the-available-bioassays" title="Link to this heading">#</a></h3>
<p>Recall from the Bioactivity Primer we found there were 3 bioassays (AIDs <a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743065">743065</a>, <a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743066">743066</a>, <a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743067">743067</a>) associated with the rat thyroid receptor β:
https://pubchem.ncbi.nlm.nih.gov/protein/NP_001257783#section=Chemicals-and-Bioactivities</p>
<ul class="simple">
<li><p><a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743065">Assay 743065 was the qHTS assay to identify small molecule antagonists of the thyroid receptor (TR) signaling pathway.</a></p></li>
<li><p><a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743066">Assay 743066 was the qHTS assay to identify small molecule agonists of the thyroid receptor (TR) signaling pathway.</a></p></li>
<li><p><a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743067">Assay 743067 was a Summary file.</a></p></li>
</ul>
<p>We can now find the unique bioassays programmatically from the downloaded CSV files in our dataframe from both the rat and human receptors.</p>
<p>First, let’s get the number of unique AIDs for rat thyroid hormone receptor.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># the .nunique() function gives the number of unique values in a dataframe column</span>
<span class="n">df_rat</span><span class="p">[</span><span class="s1">&#39;BioAssay_AID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">nunique</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3
</pre></div>
</div>
</div>
</div>
<p>Let’s get the number of data points for rat thyroid hormone receptor for each assay. These will be the total tested substances in the assay regardless of active or inactive.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_rat</span><span class="p">[</span><span class="s1">&#39;BioAssay_AID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>BioAssay_AID
743066    10486
743065    10486
743067    10486
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<ol class="arabic simple">
<li><p>Confirm that the totals above match what is in the bioassay information at Pubchem:
https://pubchem.ncbi.nlm.nih.gov/protein/NP_001257783#section=Chemicals-and-Bioactivities</p></li>
<li><p>In the code cells below check the number of unique assays and the number of data points for the human thyroid hormone receptor beta (THRβ).</p></li>
</ol>
</div><div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code for number of unique assays in human receptor</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code to determine number of assay data points (tested compounds) for each unique assay in human receptor</span>
</pre></div>
</div>
</div>
</div>
<p>Notice that there are more than one hundred assays for human THRβ. However, we are only interested in the assays with the <strong>potency</strong> values, because only the <strong>potency</strong> data is available for both rats and humans.  Therefore, we will focus on those assays.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_rat</span><span class="p">[</span><span class="n">df_rat</span><span class="p">[</span><span class="s1">&#39;Activity_Type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Potency&#39;</span><span class="p">][[</span><span class="s1">&#39;BioAssay_AID&#39;</span><span class="p">,</span> <span class="s1">&#39;BioAssay_Name&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>BioAssay_AID  BioAssay_Name                                                                                            
743065        qHTS assay to identify small molecule antagonists of the thyroid receptor (TR) signaling pathway             10486
743066        qHTS assay to identify small molecule agonists of the thyroid receptor (TR) signaling pathway                10486
743067        qHTS assay to identify small molecule antagonists of the thyroid receptor (TR) signaling pathway: Summary    10486
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-success"> 
<strong>Python Dive:</strong> The above code is creating a Boolean mask to filter the DataFrame for rows where the 'Activity_Type' is 'Potency'. It then selects the 'BioAssay_AID' and 'BioAssay_Name' columns and counts the occurrences of each unique combination. 
<div class="alert alert-block alert-info">
<details>
    <Summary>Explanation of the code</Summary>
<p><code class="docutils literal notranslate"><span class="pre">df_rat[df_rat['Activity_Type']</span> <span class="pre">==</span> <span class="pre">'Potency'][['BioAssay_AID',</span> <span class="pre">'BioAssay_Name']].value_counts()</span></code></p>
<ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">df_rat[df_rat['Activity_Type']</span> <span class="pre">==</span> <span class="pre">'Potency']</span></code> This actualy creates a mask and applies it to the dataframe, returning the filtered rows. There are two processes going on.</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">df_rat['Activity_Type']</span> <span class="pre">==</span> <span class="pre">'Potency'</span></code> - This creates a Boolean mask: a series of True/False values indicating whether each row meets the condition. We could write this as <code class="docutils literal notranslate"><span class="pre">mask</span> <span class="pre">=</span> <span class="pre">df_rat['Activity_Type']</span> <span class="pre">==</span> <span class="pre">'Potency'</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">df_rat[mask]</span></code> then filters the DataFrame and selects the rows from df_rat where the mask is True.</p></li>
</ul>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">[['BioAssay_AID',</span> <span class="pre">'BioAssay_Name']]</span></code> selects only the ‘BioAssay_AID’ and ‘BioAssay_Name’ columns from the filtered DataFrame.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.value_counts()</span></code> is a Pandas method that counts the occurrences of each unique combination of ‘BioAssay_AID’ and ‘BioAssay_Name’.</p>
<ul class="simple">
<li><p>Returns a Series with the counts of each unique combination.</p></li>
<li><p>The index is a MultiIndex with ‘BioAssay_AID’ and ‘BioAssay_Name’, this returns all observed combinations of these two columns and their counts.</p></li>
</ul>
</li>
</ol>
</details>
</div>
</div><div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your code here to filter the data frame for human receptor and count the number of assays with potency data.</span>
</pre></div>
</div>
</div>
</div>
<p>Considering the title and the data point counts for the returned assays, the three assays for rat thyroid hormone receptor are high-throughput screening (HTS) assays.  Therefore, we also need HTS assays for human thyroid hormone receptor.</p>
<p>In the remaining part of this notebook, we will use the qHTS assays for antagonists of rat and human thyroid hormone receptors.</p>
<ul class="simple">
<li><p><a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743065"><strong>AID 743065</strong></a>: Confirmatory Assay (Antagonists)  in rat receptor</p></li>
<li><p><a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/588547"><strong>AID 588547</strong></a>: Confirmatory (Antagonists)  in human receptor</p></li>
</ul>
<p>They were chosen based on the following considerations.</p>
<ul class="simple">
<li><p>The four assays for human thyroid hormone receptors with more than 1000 data points (i.e., 1469, 1479, 588547, 588545) are HTS screening.</p></li>
<li><p>Strictly speaking, AIDs 1469 and 1479 are not targeting the human thyroid hormone receptor, but its interaction with steroid receptor coregulator.</p></li>
<li><p>At the <a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743067#section=Comment">summary page of <strong>AID 743067</strong></a>, it is explained that the assay summarizes from an confirmatory antagonist assay (<a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743065"><strong>AID 743065</strong></a>) and a cell viability counter screening (<a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743064"><strong>AID 743064</strong></a>).</p></li>
<li><p>More active compounds were identified in the antagonist HTS assays for both species (<a class="reference external" href="https::/pubchem.ncbi.nlm.nih.gov/bioassay/743065"><strong>AIDs 743065</strong></a> and <a class="reference external" href="https::/pubchem.ncbi.nlm.nih.gov/bioassay/588547"><strong>588547</strong></a>, compared to the agonist assays (<a class="reference external" href="https::/pubchem.ncbi.nlm.nih.gov/bioassay/743066"><strong>AIDs 743066</strong></a> and <a class="reference external" href="https::/pubchem.ncbi.nlm.nih.gov/bioassay/588545"><strong>588545</strong></a>).</p></li>
</ul>
<center>
<div class="pst-scrollable-table-container"><table class="table">
<thead>
<tr class="row-odd"><th class="head"><p>AID</p></th>
<th class="head"><p>Assay Type</p></th>
<th class="head text-center"><p>species</p></th>
<th class="head text-center"><p>Number of tested substances</p></th>
<th class="head text-center"><p>number of actives</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><strong>743065</strong></p></td>
<td><p><span style="color:red">Antagonist</span></p></td>
<td class="text-center"><p>rat 🐀</p></td>
<td class="text-center"><p>10,486</p></td>
<td class="text-center"><p>1880</p></td>
</tr>
<tr class="row-odd"><td><p><strong>588547</strong></p></td>
<td><p><span style="color:red">Antagonist</span></p></td>
<td class="text-center"><p>human 🧑🏽‍🔬</p></td>
<td class="text-center"><p>2,858</p></td>
<td class="text-center"><p>12</p></td>
</tr>
<tr class="row-even"><td><p><strong>743066</strong></p></td>
<td><p><span style="color:green">Agonist</span></p></td>
<td class="text-center"><p>rat 🐀</p></td>
<td class="text-center"><p>10,486</p></td>
<td class="text-center"><p>64</p></td>
</tr>
<tr class="row-odd"><td><p><strong>588545</strong></p></td>
<td><p><span style="color:green">Agonist</span></p></td>
<td class="text-center"><p>human🧑🏽‍🔬</p></td>
<td class="text-center"><p>2,825</p></td>
<td class="text-center"><p>10</p></td>
</tr>
</tbody>
</table>
</div>
</section>
</section>
<section id="comparing-hts-screening-data-for-rat-and-human">
<h2>Comparing HTS screening data for Rat and human<a class="headerlink" href="#comparing-hts-screening-data-for-rat-and-human" title="Link to this heading">#</a></h2>
<p>The following section uses the quantitative HTS (qHTS) data for thyroid hormone receptor antagonsts.  The data files were downloaded from their BioAssay record pages: (Note: Links below take you to the Download csv dropdown in the data table.)</p>
<ul class="simple">
<li><p><a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/743065#section=Result-Definitions:~:text=10%2C486%20test%20results-,Download,-SORT%20BY"><strong>AID 743065</strong></a>: Confirmatory Assay (Antagonists)</p></li>
<li><p><a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/bioassay/588547#section=Data-Table:~:text=2%2C858%20test%20results-,Download,-SORT%20BY"><strong>AID 588547</strong></a>: Confirmatory (antagonists)</p></li>
</ul>
<p>We have provided you with the data files as compressed ‘zip’ files, but you can also download them directly as .csv files. The names of the downloaded files are:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#file_aid_rat = &quot;AID_743065_datatable_all.csv&quot;  #uncomment to use uncompressed csv files</span>
<span class="c1">#file_aid_human = &quot;AID_588547_datatable_all.csv&quot; #uncomment to use uncompressed csv files</span>

<span class="n">file_aid_rat</span> <span class="o">=</span> <span class="s2">&quot;AID_743065_datatable_all.zip&quot;</span>
<span class="n">file_aid_human</span> <span class="o">=</span> <span class="s2">&quot;AID_588547_datatable_all.zip&quot;</span>
</pre></div>
</div>
</div>
</div>
<section id="loading-the-hts-screening-data-for-rats">
<h3>Loading the HTS screening data for Rats<a class="headerlink" href="#loading-the-hts-screening-data-for-rats" title="Link to this heading">#</a></h3>
<p>Now let’s load the HTS data for the rat thyroid hormone receptor.</p>
<ul class="simple">
<li><p>The data type for the <code class="docutils literal notranslate"><span class="pre">PUBCHEM_CID</span></code> column is explicitly specified <code class="docutils literal notranslate"><span class="pre">mydtype</span> <span class="pre">=</span> <span class="pre">{'PUBCHEM_CID':'Int64'</span> <span class="pre">}</span></code> to ensure it is read as an integer, which is important for accurate data handling.</p></li>
<li><p>The first 5 rows are skipped <code class="docutils literal notranslate"><span class="pre">skiprows=[1,2,3,4,5]</span></code> because they do not contain data, but meta data (e.g., the description of each column).</p></li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mydtype</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">:</span><span class="s1">&#39;Int64&#39;</span> <span class="p">}</span>
<span class="n">df_aid_rat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file_aid_rat</span><span class="p">,</span> <span class="n">compression</span><span class="o">=</span><span class="s1">&#39;zip&#39;</span><span class="p">,</span> <span class="n">skiprows</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">5</span><span class="p">],</span> <span class="n">dtype</span><span class="o">=</span><span class="n">mydtype</span><span class="p">,</span> <span class="n">low_memory</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>PUBCHEM_RESULT_TAG</th>
      <th>PUBCHEM_SID</th>
      <th>PUBCHEM_CID</th>
      <th>PUBCHEM_EXT_DATASOURCE_SMILES</th>
      <th>PUBCHEM_ACTIVITY_OUTCOME</th>
      <th>PUBCHEM_ACTIVITY_SCORE</th>
      <th>PUBCHEM_ACTIVITY_URL</th>
      <th>PUBCHEM_ASSAYDATA_COMMENT</th>
      <th>Phenotype-Replicate_1</th>
      <th>Potency-Replicate_1</th>
      <th>...</th>
      <th>Activity at 0.762 uM-Replicate_46</th>
      <th>Activity at 1.702 uM-Replicate_46</th>
      <th>Activity at 3.793 uM-Replicate_46</th>
      <th>Activity at 8.439 uM-Replicate_46</th>
      <th>Activity at 18.15 uM-Replicate_46</th>
      <th>Activity at 32.49 uM-Replicate_46</th>
      <th>Activity at 70.35 uM-Replicate_46</th>
      <th>Activity at 94.96 uM-Replicate_46</th>
      <th>Activity at 260.0 uM-Replicate_46</th>
      <th>Activity at 920.3 uM-Replicate_46</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>144203552</td>
      <td>12850184</td>
      <td>C(C(=O)[C@H]([C@@H]([C@H](C(=O)[O-])O)O)O)O.C(...</td>
      <td>Inactive</td>
      <td>0</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>NaN</td>
      <td>Inactive</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2</td>
      <td>144203553</td>
      <td>89753</td>
      <td>C([C@H]([C@H]([C@@H]([C@H](C(=O)[O-])O)O)O)O)O...</td>
      <td>Inactive</td>
      <td>0</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>NaN</td>
      <td>Inactive</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3</td>
      <td>144203554</td>
      <td>9403</td>
      <td>C[C@]12CC[C@H]3[C@H]([C@@H]1CC[C@@H]2OC(=O)CCC...</td>
      <td>Inconclusive</td>
      <td>10</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>NaN</td>
      <td>Inhibitor</td>
      <td>33.4915</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4</td>
      <td>144203555</td>
      <td>13218779</td>
      <td>C[C@@]12CC[C@@H](C1(C)C)C[C@H]2OC(=O)CSC#N</td>
      <td>Inactive</td>
      <td>0</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>NaN</td>
      <td>Inactive</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>144203556</td>
      <td>142766</td>
      <td>C1=CC=C(C(=C1)C(=O)O)O.C1=CC2=C(C(=C1)O)N=CC=C2</td>
      <td>Active</td>
      <td>83</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>NaN</td>
      <td>Inhibitor</td>
      <td>8.4127</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 1803 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(10486, 1803)
</pre></div>
</div>
</div>
</div>
<p>This code cell identifies and counts all columns that do not have values.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">nan_columns</span> <span class="o">=</span> <span class="n">df_aid_rat</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">isna</span><span class="p">()</span><span class="o">.</span><span class="n">all</span><span class="p">()]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">nan_columns</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>389
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-success"> 
<strong>Python Dive:</strong> The above code tells us there are 389 columns with no values. 
<div class="alert alert-block alert-info">
<details>
    <Summary>Explanation of the code</Summary>
<p><code class="docutils literal notranslate"><span class="pre">nan_columns</span> <span class="pre">=</span> <span class="pre">df_aid_rat.columns[df_aid_rat.isna().all()].tolist()</span></code></p>
<ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">df_aid_rat.isna()</span></code>  Returns a DataFrame of the same shape</p>
<ul class="simple">
<li><p>Every cell is <code class="docutils literal notranslate"><span class="pre">True</span></code> if NaN, otherwise <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
</ul>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">.all()</span></code> → Checks if all values in each column are True (i.e., all values are NaN).</p>
<ul class="simple">
<li><p>applied column-wise (by default axis=0), so it goes down each column to see if it is true, so there is one Boolean value for each column.</p></li>
<li><p>the result is a Pandas Series of Booleans, where the index is the name of the column, and each value is a Boolean True or False based on the values of the entire column.</p></li>
</ul>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">df_aid_rat.columns[...]</span></code> Filters the DataFrame’s columns (a Pandas Series) to keep only those where the condition is True.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.tolist()</span></code> → Converts the filtered column names to a list.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">nan_columns</span> <span class="pre">=</span> <span class="pre">df_aid_rat.columns[df_aid_rat.isna().all()].tolist()</span></code>  Assigns the list of column names with all NaN values to the variable <code class="docutils literal notranslate"><span class="pre">nan_columns</span></code>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">len(nan_columns)</span></code>  Returns the number of columns that have all values as NaN.</p></li>
<li><p>print(len(nan_columns))  Prints the number of columns with all NaN values.</p></li>
</ol>
</details>
</div>
</div><p>The following cell removes those columns and displays the updated dimensions.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span> <span class="o">=</span> <span class="n">df_aid_rat</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">how</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">df_aid_rat</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(10486, 1414)
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-success"> 
<strong>Python Dive:</strong> The above code is an important step in cleaning data where you drop all values that are *Not a Number* and we reduced our columns from 1803 to 1414. 
<div class="alert alert-block alert-info">
<details>
    <Summary>Explanation of the code</Summary>
<p><code class="docutils literal notranslate"><span class="pre">df_aid_rat</span> <span class="pre">=</span> <span class="pre">df_aid_rat.dropna(how='all',</span> <span class="pre">axis=1)</span></code></p>
<ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">.dropna()</span></code>is a Pandas method that removes missing values (NaN) from a DataFrame.</p>
<ul class="simple">
<li><p>it only looks for cells with missing data (NaN), not for strings like “active” or “CCO”. A column with text values will not be dropped unless every single value in it is missing (NaN)</p></li>
</ul>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">how='all'</span></code> specifies that only columns where all values are NaN should be dropped.</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">how</span></code> is a keyword parameter and ‘all’ is the argument that you pass to it.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">axis</span></code> is another keyword parameter that specifies whether to drop rows (axis=0) or columns (axis=1), where 0 and 1 are the arguments you pass to it</p></li>
<li><p>Quick Definitions</p>
<ul>
<li><p><strong>Function</strong> → A reusable block of code that does something when you call it. Example: <code class="docutils literal notranslate"><span class="pre">print(&quot;hello&quot;)</span></code>.</p></li>
<li><p><strong>Method</strong> → A function that <em>belongs to an object</em> and is called using dot notation. Example: <code class="docutils literal notranslate"><span class="pre">df.dropna()</span></code>.</p></li>
<li><p><strong>Parameter</strong> → The <em>placeholder names</em> defined inside the function/method (e.g. <code class="docutils literal notranslate"><span class="pre">axis</span></code>, <code class="docutils literal notranslate"><span class="pre">how</span></code>).</p></li>
<li><p><strong>Argument</strong> → The <em>actual values</em> you pass to those parameters when you call the function/method (e.g. <code class="docutils literal notranslate"><span class="pre">axis=1</span></code>, <code class="docutils literal notranslate"><span class="pre">&quot;all&quot;</span></code>).</p></li>
<li><p><strong>Positional vs Keyword arguments</strong> →</p>
<ul>
<li><p><strong>Positional</strong>: arguments matched by their order. Example: <code class="docutils literal notranslate"><span class="pre">df.dropna(1,</span> <span class="pre">&quot;all&quot;)</span></code> (first goes to <code class="docutils literal notranslate"><span class="pre">axis</span></code>, second goes to <code class="docutils literal notranslate"><span class="pre">how</span></code>).</p></li>
<li><p><strong>Keyword</strong>: arguments matched by name, not order. Example: <code class="docutils literal notranslate"><span class="pre">df.dropna(axis=1,</span> <span class="pre">how=&quot;all&quot;)</span></code>.</p></li>
</ul>
</li>
<li><p><strong>NaN</strong> (“Not a Number”) is a special floating-point value defined by the IEEE 754 standard, used in Python to represent missing, undefined, or invalid numerical data.</p></li>
</ul>
</li>
</ul>
</li>
</ol>
</details>
</div>
</div><div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>PUBCHEM_RESULT_TAG</th>
      <th>PUBCHEM_SID</th>
      <th>PUBCHEM_CID</th>
      <th>PUBCHEM_EXT_DATASOURCE_SMILES</th>
      <th>PUBCHEM_ACTIVITY_OUTCOME</th>
      <th>PUBCHEM_ACTIVITY_SCORE</th>
      <th>PUBCHEM_ACTIVITY_URL</th>
      <th>Phenotype-Replicate_1</th>
      <th>Potency-Replicate_1</th>
      <th>Efficacy-Replicate_1</th>
      <th>...</th>
      <th>Activity at 0.153 uM-Replicate_46</th>
      <th>Activity at 0.341 uM-Replicate_46</th>
      <th>Activity at 0.762 uM-Replicate_46</th>
      <th>Activity at 1.702 uM-Replicate_46</th>
      <th>Activity at 3.793 uM-Replicate_46</th>
      <th>Activity at 8.439 uM-Replicate_46</th>
      <th>Activity at 18.15 uM-Replicate_46</th>
      <th>Activity at 32.49 uM-Replicate_46</th>
      <th>Activity at 70.35 uM-Replicate_46</th>
      <th>Activity at 94.96 uM-Replicate_46</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>144203552</td>
      <td>12850184</td>
      <td>C(C(=O)[C@H]([C@@H]([C@H](C(=O)[O-])O)O)O)O.C(...</td>
      <td>Inactive</td>
      <td>0</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>Inactive</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2</td>
      <td>144203553</td>
      <td>89753</td>
      <td>C([C@H]([C@H]([C@@H]([C@H](C(=O)[O-])O)O)O)O)O...</td>
      <td>Inactive</td>
      <td>0</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>Inactive</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3</td>
      <td>144203554</td>
      <td>9403</td>
      <td>C[C@]12CC[C@H]3[C@H]([C@@H]1CC[C@@H]2OC(=O)CCC...</td>
      <td>Inconclusive</td>
      <td>10</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>Inhibitor</td>
      <td>33.4915</td>
      <td>71.8523</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4</td>
      <td>144203555</td>
      <td>13218779</td>
      <td>C[C@@]12CC[C@@H](C1(C)C)C[C@H]2OC(=O)CSC#N</td>
      <td>Inactive</td>
      <td>0</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>Inactive</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>144203556</td>
      <td>142766</td>
      <td>C1=CC=C(C(=C1)C(=O)O)O.C1=CC2=C(C(=C1)O)N=CC=C2</td>
      <td>Active</td>
      <td>83</td>
      <td>http://assay.nih.gov/htsws/rest/display/tox21-...</td>
      <td>Inhibitor</td>
      <td>8.4127</td>
      <td>111.3000</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 1414 columns</p>
</div></div></div>
</div>
<div class="alert alert-block alert-warning">
<H3> Loading the HTS data for humans</H3>
<p>In the following code cells you will read in the human HTS data as we did for the rat.</p>
</div><p>Read the zip (or csv) file for the human HTS data into a dataframe called <code class="docutils literal notranslate"><span class="pre">df_aid_human</span></code>. Make sure to skip the initial descriptive rows and set the data type for ‘PUBCHEM_CID’ to ‘Int64’.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code here.</span>
</pre></div>
</div>
</div>
</div>
<p>Output the first 5 rows of the human AID dataframe to check that it properly loaded.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code here.</span>
</pre></div>
</div>
</div>
</div>
<p>Display the dimensions of the human HTS DataFrame.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code here.</span>
</pre></div>
</div>
</div>
</div>
<p>In the following cell remove the columns containing no values and displays the updated dimensions.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code here.</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_human</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="substance-level-analysis">
<h3>Substance-level analysis<a class="headerlink" href="#substance-level-analysis" title="Link to this heading">#</a></h3>
<p>This section compares the HTS data for rats and humans at the substance level</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">columns</span>    <span class="c1"># Uncomment this line if you want to see what columns the data frame has.</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Index([&#39;PUBCHEM_RESULT_TAG&#39;, &#39;PUBCHEM_SID&#39;, &#39;PUBCHEM_CID&#39;,
       &#39;PUBCHEM_EXT_DATASOURCE_SMILES&#39;, &#39;PUBCHEM_ACTIVITY_OUTCOME&#39;,
       &#39;PUBCHEM_ACTIVITY_SCORE&#39;, &#39;PUBCHEM_ACTIVITY_URL&#39;,
       &#39;Phenotype-Replicate_1&#39;, &#39;Potency-Replicate_1&#39;, &#39;Efficacy-Replicate_1&#39;,
       ...
       &#39;Activity at 0.153 uM-Replicate_46&#39;,
       &#39;Activity at 0.341 uM-Replicate_46&#39;,
       &#39;Activity at 0.762 uM-Replicate_46&#39;,
       &#39;Activity at 1.702 uM-Replicate_46&#39;,
       &#39;Activity at 3.793 uM-Replicate_46&#39;,
       &#39;Activity at 8.439 uM-Replicate_46&#39;,
       &#39;Activity at 18.15 uM-Replicate_46&#39;,
       &#39;Activity at 32.49 uM-Replicate_46&#39;,
       &#39;Activity at 70.35 uM-Replicate_46&#39;,
       &#39;Activity at 94.96 uM-Replicate_46&#39;],
      dtype=&#39;object&#39;, length=1414)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_human</span><span class="o">.</span><span class="n">columns</span> <span class="c1"># Uncomment this line if you want to see what columns the data frame has.</span>
</pre></div>
</div>
</div>
</div>
<section id="activity-outcomes-and-phenotypes">
<h4>Activity outcomes and phenotypes<a class="headerlink" href="#activity-outcomes-and-phenotypes" title="Link to this heading">#</a></h4>
<p>This part of the analysis focuses on comparing the activity outcomes and observed phenotypes from the HTS data for both species.</p>
<p>First, let’s generate two bar charts to visually compare the distribution of <code class="docutils literal notranslate"><span class="pre">PUBCHEM_ACTIVITY_OUTCOME</span></code> in the rat and human HTS datasets.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">countplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_rat</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sns</span><span class="o">.</span><span class="n">countplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_human</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Number of Data Points (rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Number of Data Points (human)&quot;</span><span class="p">)</span>

<span class="k">for</span> <span class="n">ax</span> <span class="ow">in</span> <span class="n">axes</span> <span class="p">:</span>    <span class="c1"># loop over each chart.</span>

    <span class="k">for</span> <span class="n">container</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">containers</span><span class="p">:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">bar_label</span><span class="p">(</span><span class="n">container</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<ol class="arabic simple">
<li><p>What is the majority activity data type in each species?</p></li>
<li><p>How many actives are found for each species?</p></li>
<li><p>Is an active molecule an agonist or antagonist in this assay? (recall each of the bioassay titles)</p></li>
</ol>
</div><p>Both data sets have a column containing information on the type of activity found for the substance tested.  These columns are:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">Phenotype-Replicate_1</span></code> for rats</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Phenotype</span></code> for humans</p></li>
</ul>
<p>It is worth mentioning that the column for rats is suffixed with <code class="docutils literal notranslate"><span class="pre">-Replicate_1</span></code>.  The rat data set contains data from repeated assay experiments, which are indicated as Replicate_1, Replicate_2, Replicate_3, and so on. For simplicity, we will use the data for <code class="docutils literal notranslate"><span class="pre">-Replicate_1</span></code>, but if we were to perform a stricter analysis, we should consider data from all replicates.</p>
<p>Now let’s create two bar charts to compare the distribution of phenotypes (‘Phenotype-Replicate_1’ for rat and ‘Phenotype’ for human) across the two HTS datasets.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>In each case the phenotype indicates the type of activity observed: inhibitor, activator, fluorescent, cytotoxic, inactive, or inconclusive.</p>
<p>To find the relationship between the activity outcomes and phenotypes, let’s count the occurrences of each unique combination of ‘PUBCHEM_ACTIVITY_OUTCOME’ and ‘Phenotype-Replicate_1’ in the <strong>rat HTS data</strong>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="p">[[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="s1">&#39;Phenotype-Replicate_1&#39;</span> <span class="p">]]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>PUBCHEM_ACTIVITY_OUTCOME  Phenotype-Replicate_1
Inactive                  Inactive                 7041
Active                    Inhibitor                1880
Inconclusive              Inhibitor                1038
Inactive                  Activator                 527
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">.groupby()</span></code> and <code class="docutils literal notranslate"><span class="pre">.size()</span></code> can be used to get the same information (but sorted differently).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="s1">&#39;Phenotype-Replicate_1&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">size</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>PUBCHEM_ACTIVITY_OUTCOME  Phenotype-Replicate_1
Active                    Inhibitor                1880
Inactive                  Activator                 527
                          Inactive                 7041
Inconclusive              Inhibitor                1038
dtype: int64
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-success"> 
<strong>Python Dive:</strong> The above code uses the pandas method groupby to group the data fields of PUBCHEM_ACTIVITY_OUTCOME and Phenotype-Replicate_1, and then uses .size() method to count the occurrences of each unique combination.
<div class="alert alert-block alert-info">
<details>
    <Summary>Explanation of the code</Summary>
<p><code class="docutils literal notranslate"><span class="pre">df_aid_rat.groupby(by=['PUBCHEM_ACTIVITY_OUTCOME',</span> <span class="pre">'Phenotype-Replicate_1']).size()</span></code></p>
<ol class="arabic simple">
<li><p><strong><code class="docutils literal notranslate"><span class="pre">groupby(by=[...])</span></code></strong></p>
<ul class="simple">
<li><p>Splits the DataFrame into groups based on the unique combinations of two columns:</p>
<ul>
<li><p><code class="docutils literal notranslate"><span class="pre">PUBCHEM_ACTIVITY_OUTCOME</span></code> (e.g., Active, Inactive, Inconclusive)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Phenotype-Replicate_1</span></code> (e.g., Activator, Inhibitor, Inactive)</p></li>
</ul>
</li>
<li><p>Each unique pair of values is a <strong>group key</strong>.</p></li>
</ul>
</li>
<li><p><strong><code class="docutils literal notranslate"><span class="pre">.size()</span></code></strong></p>
<ul class="simple">
<li><p>Counts the number of rows in each group.</p></li>
<li><p>The result is a <strong>Pandas Series</strong>, where:</p>
<ul>
<li><p>The <strong>index</strong> = the group keys (a MultiIndex of outcome + phenotype).</p></li>
<li><p>The <strong>values</strong> = the row counts for each group.</p></li>
</ul>
</li>
</ul>
</li>
</ol>
<p>The result is a Pandas Series with a MultiIndex built from the two grouping columns, with values being the counts (the output of .size()).</p>
<ul class="simple">
<li><p>Level 0 → PUBCHEM_ACTIVITY_OUTCOME</p></li>
<li><p>Level 1 → Phenotype-Replicate_1</p></li>
</ul>
<p>So internally it looks like this:</p>
<div class="highlight-ini notranslate"><div class="highlight"><pre><span></span><span class="na">index</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">MultiIndex([</span>
<span class="w">    </span><span class="na">(&#39;Active&#39;, &#39;Inhibitor&#39;),</span>
<span class="w">    </span><span class="na">(&#39;Inactive&#39;, &#39;Activator&#39;),</span>
<span class="w">    </span><span class="na">(&#39;Inactive&#39;, &#39;Inactive&#39;),</span>
<span class="w">    </span><span class="na">(&#39;Inconclusive&#39;, &#39;Inhibitor&#39;)</span>
<span class="na">], names</span><span class="o">=</span><span class="s">[&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;,&#39;Phenotype-Replicate_1&#39;])</span>

<span class="na">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">[1880, 527, 7041, 1038]</span>
</pre></div>
</div>
<p>It does not print the ‘Inactive’ label again for the second third level because it is the same as the second level, so it is not repeated in the output.</p>
</details>
</div>
<div class="alert alert-block alert-warning">
<strong>Understand the output:</strong>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="l l-Scalar l-Scalar-Plain">PUBCHEM_ACTIVITY_OUTCOME  Phenotype-Replicate_1</span>
<span class="l l-Scalar l-Scalar-Plain">Active                    Inhibitor                1880</span>
<span class="l l-Scalar l-Scalar-Plain">Inactive                  Activator                 527</span>
<span class="w">                          </span><span class="l l-Scalar l-Scalar-Plain">Inactive                 7041</span>
<span class="l l-Scalar l-Scalar-Plain">Inconclusive              Inhibitor                1038</span>
<span class="l l-Scalar l-Scalar-Plain">dtype</span><span class="p p-Indicator">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">int64</span>
</pre></div>
</div>
<ul class="simple">
<li><p><strong>Active, Inhibitor → 1880</strong> substances</p></li>
<li><p><strong>Inactive, Activator → 527</strong> substances</p></li>
<li><p><strong>Inactive, Inactive → 7041</strong> substances
<em>(the blank under “Inactive” just means “same as above” — Pandas doesn’t repeat the label)</em></p></li>
<li><p><strong>Inconclusive, Inhibitor → 1038</strong> substances</p></li>
</ul>
</div>
</div><p>Now let’s count the occurrences of each unique combination of ‘PUBCHEM_ACTIVITY_OUTCOME’ and ‘Phenotype’ in the <strong>human HTS data</strong>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_human</span><span class="p">[[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="s1">&#39;Phenotype&#39;</span> <span class="p">]]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<p>Get the same information using <code class="docutils literal notranslate"><span class="pre">.groupby()</span></code> and <code class="docutils literal notranslate"><span class="pre">.size()</span></code></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_human</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="s1">&#39;Phenotype&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">size</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<ol class="arabic simple">
<li><p>Look at the objective of the rat and human assays. What type of compounds were the assays designed to identify?</p></li>
<li><p>In your rat dataset, how are compounds with an Activator phenotype classified? How is this different from the human dataset?</p></li>
<li><p>Why do you think Activators are not declared as “Active” even though they show biological activity?</p></li>
<li><p>In the human dataset, some compounds are labeled Inconclusive. What are possible reasons a compound might be classified this way rather than as Active or Inactive?</p></li>
<li><p>The “Inconclusive” category in the human dataset includes additional phenotypes such as Cytotoxic and Fluorescent.</p>
<ul class="simple">
<li><p>What does it mean if a compound is labeled Cytotoxic?</p></li>
<li><p>Why might a fluorescent compound interfere with interpreting assay results?</p></li>
<li><p>What issues with a dose-response curve could make a compound labeled as an Inhibitor still end up in the Inconclusive category?</p></li>
<li><p>(Hint for these: We explored the test methods in the BioActivity Primier activity.)</p></li>
</ul>
</li>
<li><p>Compare the way activators are treated in the rat and human datasets. How could these differences affect your ability to compare results between species?</p></li>
<li><p>If you were the scientist in charge of designing the assay, how might you justify the choice to classify activators as Inactive versus Inconclusive?</p></li>
</ol>
</div></section>
</section>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="efficacy">
<h1>Efficacy<a class="headerlink" href="#efficacy" title="Link to this heading">#</a></h1>
<p>In this section, we compare the <strong>efficacy</strong> data from the HTS assays for both rat and human thyroid hormone receptors.</p>
<p>For agonist-mode assays, high efficacy = strong activation relative to baseline.</p>
<p>For this antagonist-mode assay, efficacy instead reflects the maximum inhibition observed relative to the agonist control (thyroid hormone T3).</p>
<ul class="simple">
<li><p>An efficacy near -100% means the compound fully blocked the receptor’s activity back to baseline (complete antagonist).</p></li>
<li><p>An efficacy near 0% means little or no inhibition.</p></li>
<li><p>Intermediate values (e.g., -40%, -60%) mean partial inhibition.</p></li>
</ul>
<p>So in this dataset, the Efficacy column tells you “how much did the compound reduce the T3 signal at its maximum effect?”</p>
<p>Let’s look at some examples from the rat dataset to get a sense of the type of data here:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cids_of_interest</span> <span class="o">=</span> <span class="p">[</span><span class="mi">6</span><span class="p">,</span> <span class="mi">142766</span><span class="p">,</span> <span class="mi">89753</span><span class="p">,</span> <span class="mi">102428</span><span class="p">,</span> <span class="mi">142766</span><span class="p">,</span>  <span class="mi">2724411</span><span class="p">,</span> <span class="mi">12850184</span><span class="p">,</span> <span class="mi">89753</span><span class="p">,</span> <span class="mi">13218779</span><span class="p">,</span> <span class="mi">774</span><span class="p">,</span> <span class="mi">5803</span><span class="p">]</span>
<span class="n">df_aid_rat</span><span class="p">[</span><span class="n">df_aid_rat</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">cids_of_interest</span><span class="p">)]</span> \
    <span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">)</span> \
    <span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="p">[</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">,</span> <span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="s1">&#39;Phenotype-Replicate_1&#39;</span><span class="p">,</span> <span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">]]</span> \
    <span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>PUBCHEM_CID</th>
      <th>PUBCHEM_ACTIVITY_OUTCOME</th>
      <th>Phenotype-Replicate_1</th>
      <th>Efficacy-Replicate_1</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>4</th>
      <td>142766</td>
      <td>Active</td>
      <td>Inhibitor</td>
      <td>111.3000</td>
    </tr>
    <tr>
      <th>7</th>
      <td>2724411</td>
      <td>Active</td>
      <td>Inhibitor</td>
      <td>105.6700</td>
    </tr>
    <tr>
      <th>5600</th>
      <td>6</td>
      <td>Active</td>
      <td>Inhibitor</td>
      <td>121.3320</td>
    </tr>
    <tr>
      <th>0</th>
      <td>12850184</td>
      <td>Inactive</td>
      <td>Inactive</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>89753</td>
      <td>Inactive</td>
      <td>Inactive</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>13218779</td>
      <td>Inactive</td>
      <td>Inactive</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>165</th>
      <td>774</td>
      <td>Inactive</td>
      <td>Activator</td>
      <td>21.8768</td>
    </tr>
    <tr>
      <th>316</th>
      <td>5803</td>
      <td>Inactive</td>
      <td>Activator</td>
      <td>49.7967</td>
    </tr>
    <tr>
      <th>1312</th>
      <td>102428</td>
      <td>Inactive</td>
      <td>Activator</td>
      <td>19.7701</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="alert alert-block alert-success"> 
<strong>Python Dive:</strong> The above code extracts your compounds of interest, removes duplicates, narrows the view to the 4 most relevant columns, and sorts them by assay outcome so you can quickly compare Active vs. Inactive compounds along with their phenotypes and efficacy values.
<div class="alert alert-block alert-info">
<details>
    <Summary>Explanation of the code</Summary>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>`df_aid_rat [&#39;PUBCHEM_CID&#39;].isin(cids_of_interest).drop_duplicates(subset=&#39;PUBCHEM_CID&#39;).loc[:, [&#39;PUBCHEM_CID&#39;, &#39;PUBCHEM_ACTIVITY_OUTCOME&#39;, &#39;Phenotype-Replicate_1&#39;, &#39;Efficacy-Replicate_1&#39;]].sort_values(by=&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;)
</pre></div>
</div>
<p>`
Note, in Python the ‘’ is used to indicate that the line continues on the next line, so you can break long lines into multiple lines for readability, and the code box is really two lines of code, one defining the list and the other what is done with it.</p>
<ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">df_aid_rat</span></code> is a Pandas Dataframe</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">df_aid_rat['PUBCHEM_CID']</span></code> is a Pandas Series of made of the PubChem_CID field of the Dataframe.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">df_aid_rat['PUBCHEM_CID'].isin(cids_of_interest)</span></code> note <code class="docutils literal notranslate"><span class="pre">isin()</span></code>is a series method operating on the CID coluns to create a Boolean mask that is True for rows where the PUBCHEM_CID is in the list of cids_of_interest.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">df_aid_rat[df_aid_rat['PUBCHEM_CID'].isin(cids_of_interest)]</span></code> creates a new filtered DataFrame that keeps row where the mask is True. It does not overwrite the original dataframe.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.drop_duplicates(subset='PUBCHEM_CID')</span></code> removes duplicate rows based on the ‘PUBCHEM_CID’ column, keeping only the first occurrence.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.loc[:,</span> <span class="pre">['PUBCHEM_CID',</span> <span class="pre">'PUBCHEM_ACTIVITY_OUTCOME',</span> <span class="pre">'Phenotype-Replicate_1',</span> <span class="pre">'Efficacy-Replicate_1']]</span></code> selects only the specified columns from the filtered DataFrame.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.sort_values(by='PUBCHEM_ACTIVITY_OUTCOME')</span></code> sorts the DataFrame by the ‘PUBCHEM_ACTIVITY_OUTCOME’ column in ascending order.</p></li>
</ol>
</details>
</div>
</div><div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<ol class="arabic simple">
<li><p>To be labeled as an active antagonist in this model, what must the efficacy value be? Which CIDs in the data subset are antagonists? How is that indicated by phenotype?</p></li>
<li><p>How does the activity of inactive compounds compare to active compounds?</p></li>
<li><p>Some inhibitors have efficacy values greater than 100%, meaning they reduce the assay signal below the baseline. What experimental or biological factors could cause an antagonist to produce a signal lower than baseline?</p></li>
<li><p>Activators such as CID 774, 5803, and 102428 only show partial inhibition of the assay signal. Could these compounds be acting as partial agonists rather than true activators or antagonists? How might partial agonists appear in this type of antagonist assay?</p></li>
</ol>
</div><p>We can see there is similar data in the human dataset:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cids_of_interest</span> <span class="o">=</span> <span class="p">[</span><span class="mi">457193</span><span class="p">,</span> <span class="mi">15910</span><span class="p">,</span> <span class="mi">443939</span><span class="p">,</span> <span class="mi">5995</span><span class="p">,</span> <span class="mi">5281575</span><span class="p">,</span><span class="mi">7529</span><span class="p">]</span>
<span class="n">df_aid_human</span><span class="p">[</span><span class="n">df_aid_human</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">cids_of_interest</span><span class="p">)]</span> \
    <span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">)</span> \
    <span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="p">[</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">,</span> <span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="s1">&#39;Phenotype&#39;</span><span class="p">,</span> <span class="s1">&#39;Efficacy&#39;</span><span class="p">]]</span> \
    <span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Plotting the data:</p>
<p>A histogram groups data into bins to display how frequently values occur withing each range. It is best used for large datasets and shows the shape of the distribution. This can show overall patterns in data, not the individual values. This can be very useful for comparing data like potency or efficacy distributions.</p>
<p>Now that we have an overview of the types of data in the rat dataset, let’s generate two histograms to visualize and compare the distribution of efficacy values (‘Efficacy-Replicate_1’ for rat and ‘Efficacy’ for human) for different activity outcomes.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>


<span class="n">color_dict</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;Active&quot;</span><span class="p">:</span> <span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="s2">&quot;Inconclusive&quot;</span><span class="p">:</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span> <span class="s2">&quot;Conflicting&quot;</span><span class="p">:</span> <span class="s2">&quot;blue&quot;</span><span class="p">,</span> <span class="s2">&quot;Inactive&quot;</span><span class="p">:</span> <span class="s2">&quot;green&quot;</span><span class="p">}</span>

<span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_rat</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">palette</span><span class="o">=</span><span class="n">color_dict</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_human</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">palette</span><span class="o">=</span><span class="n">color_dict</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, human)&quot;</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<p>Notice in the above, it is clear we can see three distinct Activity Outcome categories in the rat data. However, with the human data being a smaller set, it is challenging to see which molecules are labeled as active or inconclusive.</p>
<p>Another way to visualize this data is with a <strong>strip plot</strong>. It will show every individual data point along an axis. If two datapoints are the same, the program may provide slight “jitter” to reduce overap. This plot is best used for showing discrete observations and identifying clusters, outliers or repeated values. It can also useful for comparing disbributions across categories. In this case we are looking at efficacy by activity outcome.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">hue_order</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Active&#39;</span><span class="p">,</span>  <span class="s1">&#39;Inconclusive&#39;</span><span class="p">,</span> <span class="s1">&#39;Inactive&#39;</span><span class="p">]</span>

<span class="n">sns</span><span class="o">.</span><span class="n">stripplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_rat</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sns</span><span class="o">.</span><span class="n">stripplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_human</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, human)&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Another useful visualization tool is the <strong>box plot</strong> which summarizes the distribution of a datase by showing five key statistics:</p>
<ul class="simple">
<li><p>minimum</p></li>
<li><p>Q1 - The First quartile (25th percentile of values)</p></li>
<li><p>Q2 - The Second quartile (50th percentile of values)</p></li>
<li><p>Median- The value exactly in the middle of the values</p></li>
<li><p>Q3 - The Third quartile (75th percentile of values)</p></li>
<li><p>Maximum - The highest non-outlier value</p></li>
<li><p>Outliers - points beyond the “whiskers” that are anomolous data points.</p></li>
</ul>
<p>You can use it to summarize variables (like efficacy or potency) and how they are distributed across categories (active, inactive, inconclusive), how to compare medians and spread between groups, and spot outliers quickly without visualizing every individual point.</p>
<p>Let’s create box plots of the activity and efficacy data for rats and humans.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">hue_order</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Active&#39;</span><span class="p">,</span>  <span class="s1">&#39;Inconclusive&#39;</span><span class="p">,</span> <span class="s1">&#39;Inactive&#39;</span><span class="p">]</span>

<span class="n">sns</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_rat</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sns</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_human</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, human)&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<ol class="arabic simple">
<li><p>Based on the various plots of the data, were efficacy values determined for all compounds in the human dataset?</p></li>
<li><p>The box plots of Efficacy for the rat and human datasets show that the Inconclusive and Inactive outcomes have many more outliers than the Active outcomes.</p>
<ul class="simple">
<li><p>Why do you think the Inconclusive and Inactive groups show such a wide spread of efficacy values?</p></li>
<li><p>What does this suggest about how compounds are classified in assays compared to those labeled Active?</p></li>
</ul>
</li>
</ol>
</div></section>
<section class="tex2jax_ignore mathjax_ignore" id="potency">
<h1>Potency<a class="headerlink" href="#potency" title="Link to this heading">#</a></h1>
<p>Now visualize the potency of the tested chemicals on the rat and human thyroid hormone receptors.</p>
<p>Generate two histograms to visualize and compare the distribution of potency values for different activity outcomes in the rat and human HTS datasets.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>Generate two strip plots to visualize the potency values for different activity outcomes in the rat and human HTS datasets.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>Generate two box plots to visualize the potency data for different activity outcomes in the rat and human HTS datasets.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<ol class="arabic simple">
<li><p>Based on the various plots of the data, were potency values determined for all compounds in the human dataset?</p></li>
<li><p>When comparing the rat and human datasets, the box and strip plots for Potency show fewer outliers than those for Efficacy.</p>
<ul class="simple">
<li><p>Why do you think fewer outliers appear in the potency data?</p></li>
<li><p>What does this tell you about the type of compounds that have measurable potency values compared to those that only have efficacy values?</p></li>
</ul>
</li>
</ol>
</div><p>We can compute summary statistics such as the mean, median, minimum, and maximum for the efficacy and potency values across different activity outcomes (Active, Inactive, Inconclusive) in both the rat and human datasets. These summaries help describe how the data are distributed and how results compare between species.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">)[[</span><span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">,</span> <span class="s1">&#39;Potency-Replicate_1&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Efficacy-Replicate_1</th>
      <th>Potency-Replicate_1</th>
    </tr>
    <tr>
      <th>PUBCHEM_ACTIVITY_OUTCOME</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Active</th>
      <td>110.130054</td>
      <td>22.882127</td>
    </tr>
    <tr>
      <th>Inactive</th>
      <td>50.140904</td>
      <td>27.487615</td>
    </tr>
    <tr>
      <th>Inconclusive</th>
      <td>58.654658</td>
      <td>43.571585</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">)[[</span><span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">,</span> <span class="s1">&#39;Potency-Replicate_1&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">median</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Efficacy-Replicate_1</th>
      <th>Potency-Replicate_1</th>
    </tr>
    <tr>
      <th>PUBCHEM_ACTIVITY_OUTCOME</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Active</th>
      <td>113.98150</td>
      <td>18.83360</td>
    </tr>
    <tr>
      <th>Inactive</th>
      <td>39.19415</td>
      <td>24.44795</td>
    </tr>
    <tr>
      <th>Inconclusive</th>
      <td>49.11970</td>
      <td>33.49150</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">)[[</span><span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">,</span> <span class="s1">&#39;Potency-Replicate_1&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">min</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Efficacy-Replicate_1</th>
      <th>Potency-Replicate_1</th>
    </tr>
    <tr>
      <th>PUBCHEM_ACTIVITY_OUTCOME</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Active</th>
      <td>49.0617</td>
      <td>0.0007</td>
    </tr>
    <tr>
      <th>Inactive</th>
      <td>13.5294</td>
      <td>0.0019</td>
    </tr>
    <tr>
      <th>Inconclusive</th>
      <td>19.5822</td>
      <td>0.0033</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">)[[</span><span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">,</span> <span class="s1">&#39;Potency-Replicate_1&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Efficacy-Replicate_1</th>
      <th>Potency-Replicate_1</th>
    </tr>
    <tr>
      <th>PUBCHEM_ACTIVITY_OUTCOME</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Active</th>
      <td>246.592</td>
      <td>159.683</td>
    </tr>
    <tr>
      <th>Inactive</th>
      <td>246.847</td>
      <td>107.353</td>
    </tr>
    <tr>
      <th>Inconclusive</th>
      <td>199.769</td>
      <td>343.762</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Now it’s your turn to compute the mean, median, minimum, maximum of the efficacy and potency values for the different outcomes of the <strong>human</strong> data set.</p>
<div class="alert alert-block alert-success"> 
<strong>Python Dive:</strong> In the above coded cells Pandas is calling numpy statistics functions to compute the mean, median, minimum and maximum values for the efficacy and potency columns grouped by the activity outcome. 
<div class="alert alert-block alert-info">
<details>
    <Summary>Explanation of the code</Summary>
`df_aid_rat.groupby('PUBCHEM_ACTIVITY_OUTCOME')[['Efficacy-Replicate_1', 'Potency-Replicate_1']].mean()`
<ol class="arabic simple">
<li><p><strong><code class="docutils literal notranslate"><span class="pre">df_aid_rat</span></code></strong>
This is your Pandas DataFrame containing PubChem assay results. It has many columns, including:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">PUBCHEM_ACTIVITY_OUTCOME</span></code> (Inactive, Active, Inconclusive, etc.)</p></li>
<li><p>replicate values like <code class="docutils literal notranslate"><span class="pre">Efficacy-Replicate_1</span></code> and <code class="docutils literal notranslate"><span class="pre">Potency-Replicate_1</span></code>.</p></li>
</ul>
</li>
<li><p><strong><code class="docutils literal notranslate"><span class="pre">.groupby('PUBCHEM_ACTIVITY_OUTCOME')</span></code></strong></p>
<ul class="simple">
<li><p>Splits your DataFrame into <strong>sub-tables</strong> (groups), one for each unique value in <code class="docutils literal notranslate"><span class="pre">PUBCHEM_ACTIVITY_OUTCOME</span></code>.</p></li>
<li><p>For example:</p>
<ul>
<li><p>Group 1: all rows with <code class="docutils literal notranslate"><span class="pre">Inactive</span></code></p></li>
<li><p>Group 2: all rows with <code class="docutils literal notranslate"><span class="pre">Active</span></code></p></li>
<li><p>Group 3: all rows with <code class="docutils literal notranslate"><span class="pre">Inconclusive</span></code></p></li>
</ul>
</li>
<li><p>Each group can now be analyzed independently.</p></li>
</ul>
</li>
<li><p><strong><code class="docutils literal notranslate"><span class="pre">[['Efficacy-Replicate_1',</span> <span class="pre">'Potency-Replicate_1']]</span></code></strong></p>
<ul class="simple">
<li><p>This selects just <strong>two numeric columns</strong> out of the full DataFrame.</p></li>
<li><p>That way, the statistics are calculated only on these measures, not on all columns.</p></li>
</ul>
</li>
<li><p><strong><code class="docutils literal notranslate"><span class="pre">.mean()</span></code></strong></p>
<ul class="simple">
<li><p>Applies the <strong>arithmetic mean</strong> function (average) to each column, <em>within each group</em>.</p></li>
</ul>
</li>
</ol>
<p><strong>Statistics Methods</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">.mean()</span></code> → Calculates the average value for each group.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.median()</span></code> → Calculates the median value for each group.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.min()</span></code> → Finds the minimum value for each group.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.max()</span></code> → Finds the maximum value for each group.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.std()</span></code> → Calculates the standard deviation for each group.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.var()</span></code> → Calculates the variance for each group.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.count()</span></code> → Counts the number of non-null values in each group.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.size()</span></code> → Counts the total number of rows in each group (including NaN).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.quantile([0.25,</span> <span class="pre">0.5,</span> <span class="pre">0.75])</span></code> → Calculates the 25th, 50th (median), and 75th percentiles for each group.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.describe()</span></code> → Provides a summary of statistics (count, mean, std, min, 25%, 50%, 75%, max) for each group.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.agg()</span></code> → Allows you to apply multiple aggregation functions at once, e.g. <code class="docutils literal notranslate"><span class="pre">df.groupby('col').agg(['mean',</span> <span class="pre">'std',</span> <span class="pre">'min',</span> <span class="pre">'max'])</span></code>.</p></li>
</ul>
</details>
</div>
</div><div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code for mean</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code for median</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code for minimum</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code for maximum</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<Strong>Check your understanding</Strong>
<ol class="arabic simple">
<li><p>How would you expect the mean and median efficacy values of Active compounds to compare with those of Inactive or Inconclusive compounds? Is that portrayed in the data?</p></li>
<li><p>Did you observe that active compounds have higher average efficacy in one species than the other? What might that suggest about species differences in thyroid hormone receptor response?</p></li>
<li><p>Recall the lower the potency value, a lower amount of compound is needed to cause an effect. Are there species differences in minimum potency values?</p></li>
</ol>
</div><section id="efficacy-vs-potency">
<h2>Efficacy vs Potency<a class="headerlink" href="#efficacy-vs-potency" title="Link to this heading">#</a></h2>
<p>This section explores the relationship between the efficacy and potency of the tested compounds for both the rat and human thyroid hormone receptors.</p>
<p>The following code creates two scatter plots, one for the rat data and one for the human data, to visualize the relationship between potency and efficacy, with points colored by their activity outcome.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="n">hue_order</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Active&#39;</span><span class="p">,</span>  <span class="s1">&#39;Inconclusive&#39;</span><span class="p">,</span> <span class="s1">&#39;Inactive&#39;</span><span class="p">]</span>

<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_rat</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Potency-Replicate_1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_human</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Potency&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Efficacy&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Potency (micro-M, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Potency (micro-M, human)&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, human)&quot;</span><span class="p">)</span>

<span class="c1"># Uncomment next three lines to limit the x-axis to 200.</span>
<span class="c1">#for ax in axes :    # loop over each panel</span>
<span class="c1">#    ax.set_xlim(0, 200)</span>
<span class="c1">#    ax.set_ylim(0, 200)</span>
</pre></div>
</div>
</div>
</div>
<section id="compound-level-analysis">
<h3>Compound-level Analysis<a class="headerlink" href="#compound-level-analysis" title="Link to this heading">#</a></h3>
<p>In PubChem, substances are chemical records provided by individual data contributors and compounds are unique chemical structures extracted from substance records.</p>
<p>In many assays, substances can be viewed as “samples” of chemicals.  A chemical can be tested in multiple samples containing that chemical in a single assay. The samples are not necessary the same as each other. For example, they may be different in terms of its concentration, temperature, solvent used, etc.</p>
<p>Therefore, if you want to determine the activity of a chemical, you may need to collapse the data from multiple samples (i.e., substances) into a single data point for the chemical.</p>
<p>The code cells below checks how many unique compounds in the data set and how many of them are associated with multiple substances.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_rat_cidcount</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">df_aid_rat</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">df_human_cidcount</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">df_aid_human</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;#-- Rat data set&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;# CIDs in total                      :&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">df_rat_cidcount</span><span class="p">)</span> <span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;# CIDs associated with multiple SIDs :&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">df_rat_cidcount</span><span class="p">[</span><span class="n">df_rat_cidcount</span><span class="p">[</span><span class="s1">&#39;count&#39;</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">])</span> <span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;------------------------------------------------&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;#-- Human data set&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;# CIDs in total                      :&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">df_human_cidcount</span><span class="p">)</span> <span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;# CIDs associated with multiple SIDs :&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">df_human_cidcount</span><span class="p">[</span><span class="n">df_human_cidcount</span><span class="p">[</span><span class="s1">&#39;count&#39;</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">])</span> <span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="n">sns</span><span class="o">.</span><span class="n">countplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_rat_cidcount</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;count&#39;</span><span class="p">,</span>  <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sns</span><span class="o">.</span><span class="n">countplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_human_cidcount</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;count&#39;</span><span class="p">,</span>  <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Number of SIDs per CID&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Number of SIDs per CID&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of CIDs&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of CIDs&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Rat&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Human&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<section id="aggregating-substance-data-rats">
<h4>Aggregating substance data rats<a class="headerlink" href="#aggregating-substance-data-rats" title="Link to this heading">#</a></h4>
<p>Now we aggregate the substance data for each unique compound (identified by its CID).</p>
<p>This code groups the rat HTS data by ‘PUBCHEM_CID’ and store them in a data frame with three columns (<code class="docutils literal notranslate"><span class="pre">PUBCHEM_CID</span></code>, <code class="docutils literal notranslate"><span class="pre">Potency-Replicate_1</span></code>, and <code class="docutils literal notranslate"><span class="pre">Efficacy-Replicate_1</span></code>).  Note that some CIDs are associated with multiple SIDs, and hence, multiple activity values.  For those cases, we are going to take the average of the multiple activity values.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat_cid</span> <span class="o">=</span> <span class="n">df_aid_rat</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">numeric_only</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">df_aid_rat_cid</span> <span class="o">=</span> <span class="n">df_aid_rat_cid</span><span class="p">[[</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">,</span> <span class="s1">&#39;Potency-Replicate_1&#39;</span><span class="p">,</span> <span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat_cid</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>PUBCHEM_CID</th>
      <th>Potency-Replicate_1</th>
      <th>Efficacy-Replicate_1</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>4</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>4.6248</td>
      <td>118.2025</td>
    </tr>
    <tr>
      <th>2</th>
      <td>11</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>13</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>4</th>
      <td>33</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat_cid</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(8099, 3)
</pre></div>
</div>
</div>
</div>
<p>This code defines a function, <code class="docutils literal notranslate"><span class="pre">get_cid_outcome</span></code>, that determines a single activity outcome for each unique compound ID (CID) in a given DataFrame. If a compound has multiple, conflicting outcomes recorded, it is labeled as “Conflicting”.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">get_cid_outcome</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="p">:</span>

    <span class="n">outcomes</span> <span class="o">=</span> <span class="p">{}</span>

    <span class="k">for</span> <span class="n">cid</span> <span class="ow">in</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span> <span class="p">:</span>   <span class="c1"># loop over unique CIDs (without Null)</span>

        <span class="n">outcome</span> <span class="o">=</span><span class="s2">&quot;&quot;</span>
    
        <span class="n">unique_outcomes</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">cid</span> <span class="p">][</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
    
        <span class="k">if</span> <span class="p">(</span> <span class="nb">len</span><span class="p">(</span><span class="n">unique_outcomes</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span> <span class="p">)</span> <span class="p">:</span>
            <span class="n">outcome</span> <span class="o">=</span> <span class="n">unique_outcomes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
        <span class="k">elif</span> <span class="p">(</span> <span class="nb">len</span><span class="p">(</span><span class="n">unique_outcomes</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span> <span class="p">)</span> <span class="p">:</span>
            <span class="n">outcome</span> <span class="o">=</span> <span class="s2">&quot;Conflicting&quot;</span>
        <span class="k">else</span> <span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Warning: Something is wrong&quot;</span><span class="p">,</span> <span class="n">cid</span><span class="p">,</span> <span class="n">unique_outcomes</span><span class="p">)</span>

        <span class="n">outcomes</span><span class="p">[</span> <span class="n">cid</span> <span class="p">]</span> <span class="o">=</span> <span class="n">outcome</span>

    <span class="n">df_outcomes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="o">.</span><span class="n">from_dict</span><span class="p">(</span><span class="n">outcomes</span><span class="p">,</span> <span class="n">orient</span><span class="o">=</span><span class="s1">&#39;index&#39;</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">names</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">)</span>

    <span class="k">return</span><span class="p">(</span><span class="n">df_outcomes</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>This line of code applies the get_cid_outcome function to the rat HTS data to generate a DataFrame that summarizes the activity outcome for each unique compound.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_cid_outcome_rat</span> <span class="o">=</span> <span class="n">get_cid_outcome</span><span class="p">(</span><span class="n">df_aid_rat</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_cid_outcome_rat</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>PUBCHEM_CID</th>
      <th>PUBCHEM_ACTIVITY_OUTCOME</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>12850184</td>
      <td>Inactive</td>
    </tr>
    <tr>
      <th>1</th>
      <td>89753</td>
      <td>Inactive</td>
    </tr>
    <tr>
      <th>2</th>
      <td>9403</td>
      <td>Inconclusive</td>
    </tr>
    <tr>
      <th>3</th>
      <td>13218779</td>
      <td>Inactive</td>
    </tr>
    <tr>
      <th>4</th>
      <td>142766</td>
      <td>Active</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_cid_outcome_rat</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(8099, 2)
</pre></div>
</div>
</div>
</div>
<p>Now let’s merge the aggregated rat compound data (containing mean potency and efficacy) with the summarized compound outcomes.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat_cid</span> <span class="o">=</span> <span class="n">df_aid_rat_cid</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">df_cid_outcome_rat</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat_cid</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>PUBCHEM_CID</th>
      <th>Potency-Replicate_1</th>
      <th>Efficacy-Replicate_1</th>
      <th>PUBCHEM_ACTIVITY_OUTCOME</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>4</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Inactive</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>4.6248</td>
      <td>118.2025</td>
      <td>Active</td>
    </tr>
    <tr>
      <th>2</th>
      <td>11</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Inactive</td>
    </tr>
    <tr>
      <th>3</th>
      <td>13</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Inactive</td>
    </tr>
    <tr>
      <th>4</th>
      <td>33</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Inactive</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>This cell counts the occurrences of each unique ‘PUBCHEM_ACTIVITY_OUTCOME’ in the aggregated rat compound data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat_cid</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>PUBCHEM_ACTIVITY_OUTCOME
Inactive        5745
Active          1383
Inconclusive     690
Conflicting      281
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
<section id="aggregating-substance-data-for-humans">
<h4>Aggregating substance data for humans<a class="headerlink" href="#aggregating-substance-data-for-humans" title="Link to this heading">#</a></h4>
<p>Now let’s repeat the data aggregation process, this time for the human thyroid hormone receptor HTS data.</p>
<p>Group the human HTS data by ‘PUBCHEM_CID’ and store the mean efficacy and potency values in a data frame called ‘df_aid_human_cid’.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>Show a first few line of the <code class="docutils literal notranslate"><span class="pre">df_aid_human_cid</span></code> data frame.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>Use the <code class="docutils literal notranslate"><span class="pre">get_cid_outcome</span></code> function to the human HTS data to generate a DataFrame that summarizes the activity outcome for each unique compound.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>This command displays the first few rows of the DataFrame containing the summarized compound outcomes for the human data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>This code merges the aggregated human compound data with the summarized compound outcomes.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>This command displays the first few rows of the merged human compound DataFrame.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>Use <code class="docutils literal notranslate"><span class="pre">.value_counts()</span></code> the occurrences of each unique ‘PUBCHEM_ACTIVITY_OUTCOME’ in the aggregated human compound data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="merging-rat-data-and-human-data">
<h4>Merging rat data and human data<a class="headerlink" href="#merging-rat-data-and-human-data" title="Link to this heading">#</a></h4>
<p>This subsection focuses on combining the aggregated compound-level data from both the rat and human HTS assays to facilitate a direct comparison.</p>
<p>This command displays the column names of the aggregated rat compound DataFrame.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat_cid</span><span class="o">.</span><span class="n">columns</span>
</pre></div>
</div>
</div>
</div>
<p>Let’s rename the columns in the rat compound DataFrame to have consistent naming with the human compound DataFrame before merging.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat_cid</span> <span class="o">=</span> <span class="n">df_aid_rat_cid</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;Potency-Replicate_1&#39;</span><span class="p">:</span><span class="s1">&#39;Potency&#39;</span> <span class="p">,</span> <span class="s1">&#39;Efficacy-Replicate_1&#39;</span><span class="p">:</span><span class="s1">&#39;Efficacy&#39;</span><span class="p">,</span> <span class="p">})</span>
</pre></div>
</div>
</div>
</div>
<p>This command displays the first three rows of the renamed rat compound DataFrame to verify the column name changes.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_rat_cid</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Now merge the aggregated rat and human compound DataFrames based on ‘PUBCHEM_CID’, keeping only the compounds that are present in both datasets.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_merged_cid</span> <span class="o">=</span> <span class="n">df_aid_rat_cid</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">df_aid_human_cid</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_CID&#39;</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">,</span> <span class="n">suffixes</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;_rat&#39;</span><span class="p">,</span><span class="s1">&#39;_human&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<p>Let’s check if the merged data frame contains desired information.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_merged_cid</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_merged_cid</span>
</pre></div>
</div>
</div>
</div>
<p>This cell counts the occurrences of each unique ‘PUBCHEM_ACTIVITY_OUTCOME’ for the rat data within the merged DataFrame.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_merged_cid</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_rat&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<p>This cell counts the occurrences of each unique ‘PUBCHEM_ACTIVITY_OUTCOME’ for the human data within the merged DataFrame.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_merged_cid</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_human&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="compound-level-comparison">
<h4>Compound-level comparison<a class="headerlink" href="#compound-level-comparison" title="Link to this heading">#</a></h4>
<p>Let’s compare the bioactivity of the same compounds between the rat and human thyroid hormone receptors.</p>
<p>The following code cell generates two box plots to compare the distribution of efficacy for different activity outcomes between the rat and human data in the merged dataset.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">hue_order</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Active&#39;</span><span class="p">,</span>  <span class="s1">&#39;Inactive&#39;</span><span class="p">,</span> <span class="s1">&#39;Conflicting&#39;</span><span class="p">,</span> <span class="s1">&#39;Inconclusive&#39;</span><span class="p">]</span>

<span class="n">sns</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_merged_cid</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy_rat&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_rat&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_rat&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sns</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_merged_cid</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy_human&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_human&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_human&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, human)&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;PUBCHEM_ACTIVITY_OUTCOME&quot;</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<p>Generate two box plots to compare the distribution of potency for different activity outcomes between the rat and human data in the merged dataset.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your own code</span>
</pre></div>
</div>
</div>
</div>
<p>This code creates two scatter plots to visualize the relationship between potency and efficacy for the rat and human data in the merged dataset.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="n">color_dict</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;Active&quot;</span><span class="p">:</span> <span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="s2">&quot;Inconclusive&quot;</span><span class="p">:</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span> <span class="s2">&quot;Conflicting&quot;</span><span class="p">:</span> <span class="s2">&quot;blue&quot;</span><span class="p">,</span> <span class="s2">&quot;Inactive&quot;</span><span class="p">:</span> <span class="s2">&quot;green&quot;</span><span class="p">}</span>



<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_merged_cid</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Potency_rat&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Efficacy_rat&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_rat&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">palette</span><span class="o">=</span><span class="n">color_dict</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_merged_cid</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Potency_human&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Efficacy_human&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_human&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">palette</span><span class="o">=</span><span class="n">color_dict</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Potency (micro-M, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Potency (micro-M, human)&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, human)&quot;</span><span class="p">)</span>

<span class="c1"># Uncomment the next three lines to limit the x and y axes to 200.</span>
<span class="c1">#for ax in axes :    # loop over each panel</span>
<span class="c1">#    ax.set_xlim(0, 200)</span>
<span class="c1">#    ax.set_ylim(0, 200)</span>
</pre></div>
</div>
</div>
</div>
<p>The above graph output is challenging to interpret. Another useful tool for visualizing the data is a heatmap. In a heatmap, each cell’s color corresponds to the magnitude of a value. This makes it easier to see patterns, trends or correllations at a glance. They are particularly useful for comparing multiple variable (such as potency or efficacy in differen species) because you can quickly see clusters of high or low values.</p>
<p>Now we will generate a heatmap that shows the counts of rat and human activity outcome pairs. (For example, how many chemicals are active in both species, and how many chemicals are active in one species and inactive in the other?).</p>
<p>To do that we want to get the outcome pair count information from the <code class="docutils literal notranslate"><span class="pre">df_aid_merged_cid</span></code> data frame.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">outcome_counts</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">df_aid_merged_cid</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_human&#39;</span><span class="p">,</span> <span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_rat&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">size</span><span class="p">(),</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Count&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">outcome_counts</span>
</pre></div>
</div>
</div>
</div>
<p>Then, use <code class="docutils literal notranslate"><span class="pre">.pivot()</span></code> to change the ‘outcome_counts’ DataFrame into a square matrix form, where the rows are human activity outcomes, the columns are rat activity outcomes, and the values are the counts of compounds.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">heatmap_data</span> <span class="o">=</span> <span class="n">outcome_counts</span><span class="o">.</span><span class="n">pivot</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_human&#39;</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_rat&#39;</span><span class="p">,</span> <span class="n">values</span><span class="o">=</span><span class="s1">&#39;Count&#39;</span><span class="p">)</span>
<span class="n">heatmap_data</span>
</pre></div>
</div>
</div>
</div>
<p>Generate a heatmap to visualize the activity outcome pair counts.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sns</span><span class="o">.</span><span class="n">heatmap</span><span class="p">(</span><span class="n">heatmap_data</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="s2">&quot;flare&quot;</span><span class="p">,</span> <span class="n">annot</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;4.0f&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>This code creates two scatter plots to directly compare the efficacy and potency of compounds between the rat and human assays.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="n">hue_order</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Active&#39;</span><span class="p">,</span>  <span class="s1">&#39;Inactive&#39;</span><span class="p">,</span> <span class="s1">&#39;Inconclusive&#39;</span><span class="p">,</span> <span class="s1">&#39;Conflicting&#39;</span><span class="p">]</span>

<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_merged_cid</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy_rat&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Efficacy_human&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_rat&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">legend</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_merged_cid</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Potency_rat&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Potency_human&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_rat&#39;</span><span class="p">,</span> <span class="n">hue_order</span><span class="o">=</span><span class="n">hue_order</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, human)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Efficacy&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Potency (micro-M, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Potency (micro-M, human)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Potency&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.05</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper left&#39;</span><span class="p">,</span> <span class="n">borderaxespad</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># Uncomment to limit the axes to 200.</span>
<span class="c1">#for ax in axes :    # loop over each panel</span>
<span class="c1">#   ax.set_xlim(0, 200)</span>
<span class="c1">#    ax.set_ylim(0, 200)</span>
</pre></div>
</div>
</div>
</div>
<p>In the above graphs, it is not easy to see if there is any correlation between rat and human activities (in terms of the efficacy and potency).</p>
<p>Let’s focus only on the active compounds in both the rat and human assays.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_aid_merged_cid_active</span> <span class="o">=</span> <span class="n">df_aid_merged_cid</span><span class="p">[(</span><span class="n">df_aid_merged_cid</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_rat&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Active&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df_aid_merged_cid</span><span class="p">[</span><span class="s1">&#39;PUBCHEM_ACTIVITY_OUTCOME_human&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Active&#39;</span><span class="p">)]</span>
</pre></div>
</div>
</div>
</div>
<p>This code generates two scatter plots to compare the efficacy and potency of the compounds that were active in both rat and human assays.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_merged_cid_active</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Efficacy_rat&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Efficacy_human&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df_aid_merged_cid_active</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;Potency_rat&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;Potency_human&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Efficacy&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Efficacy (%, human)&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Potency&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Potency (micro-M, rat)&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Potency (micro-M, human)&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">res_efficacy</span> <span class="o">=</span> <span class="n">linregress</span><span class="p">(</span><span class="n">df_aid_merged_cid_active</span><span class="p">[</span><span class="s1">&#39;Efficacy_rat&#39;</span><span class="p">],</span> <span class="n">df_aid_merged_cid_active</span><span class="p">[</span><span class="s1">&#39;Efficacy_human&#39;</span><span class="p">])</span>
<span class="n">res_potency</span> <span class="o">=</span> <span class="n">linregress</span><span class="p">(</span><span class="n">df_aid_merged_cid_active</span><span class="p">[</span><span class="s1">&#39;Potency_rat&#39;</span><span class="p">],</span> <span class="n">df_aid_merged_cid_active</span><span class="p">[</span><span class="s1">&#39;Potency_human&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;                      Efficacy</span><span class="se">\t</span><span class="s2">Potency&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Slope               : </span><span class="si">{</span><span class="n">res_efficacy</span><span class="o">.</span><span class="n">slope</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="se">\t</span><span class="si">{</span><span class="n">res_potency</span><span class="o">.</span><span class="n">slope</span><span class="si">:</span><span class="s2">&lt;4.4f</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;intercept           : </span><span class="si">{</span><span class="n">res_efficacy</span><span class="o">.</span><span class="n">intercept</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="se">\t</span><span class="si">{</span><span class="n">res_potency</span><span class="o">.</span><span class="n">intercept</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;r-value             : </span><span class="si">{</span><span class="n">res_efficacy</span><span class="o">.</span><span class="n">rvalue</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="se">\t</span><span class="si">{</span><span class="n">res_potency</span><span class="o">.</span><span class="n">rvalue</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;p-value             : </span><span class="si">{</span><span class="n">res_efficacy</span><span class="o">.</span><span class="n">pvalue</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="se">\t</span><span class="si">{</span><span class="n">res_potency</span><span class="o">.</span><span class="n">pvalue</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;std-err (slope)     : </span><span class="si">{</span><span class="n">res_efficacy</span><span class="o">.</span><span class="n">stderr</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="se">\t</span><span class="si">{</span><span class="n">res_potency</span><span class="o">.</span><span class="n">stderr</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;std-err (intercept) : </span><span class="si">{</span><span class="n">res_efficacy</span><span class="o">.</span><span class="n">intercept_stderr</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="se">\t</span><span class="si">{</span><span class="n">res_potency</span><span class="o">.</span><span class="n">intercept_stderr</span><span class="si">:</span><span class="s2">4.4f</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>When we consider the compounds that are active in both species, there appears to be a positive correlation between the rat and human data.  However,  there are only six compounds active in both species, we cannot confidently say there is a definitive relationship.</p>
<div class="alert alert-block alert-info">
<h1>Homework</h1>
</div><p>Answer the following questions, based on the analysis of the compounds tested in both rat and human thyroid hormone receptors (i.e., the compounds in the <code class="docutils literal notranslate"><span class="pre">df_aid_merged_cid</span></code> data frame).</p>
<p>(1) How many compounds were tested against both rat thyroid hormone receptor?</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Provide your answer here</span>
</pre></div>
</div>
</div>
</div>
<p>(2) Among the compounds tested in both species, how many were active against <strong>rat</strong> thyroid hormone receptor?</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Provide your answer here</span>
</pre></div>
</div>
</div>
</div>
<p>(3) Among the compounds active in rats (i.e., those in (2)), how many were active against <strong>human</strong> thyroid hormone receptor?</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Provide your answer here</span>
</pre></div>
</div>
</div>
</div>
<p>(4) Read the following paper (open access) on the concordance of the toxicity of drugs between animals and humans and summarize its main conclusion.</p>
<p><strong>Limitations of Animal Studies for Predicting Toxicity in Clinical Trials: Is it Time to Rethink Our Current Approach?</strong><br />
Van Norman GA.<br />
JACC Basic Transl Sci. 2019 Nov 25;4(7):845-854.<br />
doi: <a class="reference external" href="https://doi.org/10.1016/j.jacbts.2019.10.008">10.1016/j.jacbts.2019.10.008</a>. PMID: 31998852; PMCID: PMC6978558.</p>
<p># Write your summary in this markdown cell:</p>
</section>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "nbi"
        },
        kernelOptions: {
            name: "nbi",
            path: "./content/modules/06-BioAssay"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'nbi'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="06-1-Bioactivity_Primer.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">6.1 Bioactivity Primer</p>
      </div>
    </a>
    <a class="right-next"
       href="06-3-eda-pathway.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Exploratory Data Analysis 2:<br><br>Chemical Modulation of Renin-Angiotensin-Aldosterone system (RAAS)</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">6.1 Exploratory Data Analysis - Protein</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#exploratory-data-analysis-1-concordance-of-thyroid-hormone-receptor-activity-in-rats-and-humans">Exploratory Data Analysis 1: <br/><br/> Concordance of Thyroid Hormone Receptor Activity β in Rats and Humans</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#checking-the-amount-of-available-data">Checking the amount of available data</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#loading-data-for-rat-thyroid-hormone-receptor">Loading data for rat thyroid hormone receptor</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#checking-the-type-and-amount-of-available-data-for-each-species">Checking the type and amount of available data for each species</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#reviewing-the-available-bioassays">Reviewing the available bioassays</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#comparing-hts-screening-data-for-rat-and-human">Comparing HTS screening data for Rat and human</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#loading-the-hts-screening-data-for-rats">Loading the HTS screening data for Rats</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#substance-level-analysis">Substance-level analysis</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#activity-outcomes-and-phenotypes">Activity outcomes and phenotypes</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#efficacy">Efficacy</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#potency">Potency</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#efficacy-vs-potency">Efficacy vs Potency</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#compound-level-analysis">Compound-level Analysis</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#aggregating-substance-data-rats">Aggregating substance data rats</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#aggregating-substance-data-for-humans">Aggregating substance data for humans</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#merging-rat-data-and-human-data">Merging rat data and human data</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#compound-level-comparison">Compound-level comparison</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Robert Belford
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>