
<!DOCTYPE html>


<html lang="en" data-content_root="../../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Fingerprint Generation &#8212; Data Chemistry 2025</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../../../_static/oldcustom.css?v=e52fa269" />
    <link rel="stylesheet" type="text/css" href="../../../_static/custom.css?v=ba2d5b83" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'content/modules/08-Molecular-Similarity/08-1-MolSimularity';</script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="What is a bit vector?" href="08-2-MolSimularity.html" />
    <link rel="prev" title="Module 8: Molecular Similarity" href="README.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../../_static/logo.png" class="logo__image only-light" alt="Data Chemistry 2025 - Home"/>
    <script>document.write(`<img src="../../../_static/logo.png" class="logo__image only-dark" alt="Data Chemistry 2025 - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../index.html">
                    2025 Data Chemistry OLCC
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../01-OLCC-Primer/README.html">Module 1: OLCC Primer</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../01-OLCC-Primer/01-1-OLCC-Primer.html">Module 1.1: OLCC Primer</a></li>







<li class="toctree-l2"><a class="reference internal" href="../01-OLCC-Primer/01-2-python-basics.html">1.2 Python Basics</a></li>


<li class="toctree-l2"><a class="reference internal" href="../01-OLCC-Primer/01-3-python-intermediate.html">1.3 Intermediate Python</a></li>


</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../02-Introduction/README.html">Module 2: Introduction</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../02-Introduction/02-1-PubChemDataTypes.html">2.1 PubChem Data Types</a></li>



<li class="toctree-l2"><a class="reference internal" href="../02-Introduction/02-2-PUG-REST-activity.html">2.2 PUG-RESTY Activity</a></li>


</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../03-Public-Chemical-Databases/README.html">Module 3: Public Chemical Databases</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../03-Public-Chemical-Databases/03-01-Survey-Public-Chemical-Databases.html">Module 1.1: Public Compound Databases</a></li>



</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../04-Chemical-Representations/README.html">Module 4: Chemical Representations</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../04-Chemical-Representations/04-01-introduction_chemical_representations.html">4.1 Chemical Representation and Introduction to RDKit</a></li>







<li class="toctree-l2"><a class="reference internal" href="../04-Chemical-Representations/04-02-PubChem3D_update.html">4.2 PubChem 3D</a></li>


<li class="toctree-l2"><a class="reference internal" href="../04-Chemical-Representations/04-03-Substructure_Searching_with_SMARTS.html">4.3 SMARTS Substructure Searching</a></li>


</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../05-Searching-Chemical-Databases/README.html">Module 5: Searching Chemical Databases</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../05-Searching-Chemical-Databases/05-01-Exploring-PubChem.html">5.1 Exploring PubChem</a></li>



<li class="toctree-l2"><a class="reference internal" href="../05-Searching-Chemical-Databases/05-02-structure-inputs.html">5.2 Searching with Structures</a></li>



<li class="toctree-l2"><a class="reference internal" href="../05-Searching-Chemical-Databases/05-03-list-conversion.html">5.3 List Conversions</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../06-BioAssay/README.html">Module 6: BioAssays</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../06-BioAssay/06-1-Bioactivity_Primer.html">6.1 Bioactivity Primer</a></li>





<li class="toctree-l2"><a class="reference internal" href="../06-BioAssay/06-2-eda-protein.html">6.1 Exploratory Data Analysis - Protein</a></li>



<li class="toctree-l2"><a class="reference internal" href="../06-BioAssay/06-3-eda-pathway.html">Exploratory Data Analysis 2:<br/><br/>Chemical Modulation of Renin-Angiotensin-Aldosterone system (RAAS)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../07-QSAR-QSPR/README.html">Module 7: Quantitative Structure Property Reationships</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../07-QSAR-QSPR/07-01-QSPR.html">Molecular Descriptors</a></li>








</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="README.html">Module 8: Molecular Similarity</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Fingerprint Generation</a></li>



<li class="toctree-l2"><a class="reference internal" href="08-2-MolSimularity.html">What is a bit vector?</a></li>


<li class="toctree-l2"><a class="reference internal" href="08-3-MolSimularity.html">Objectives</a></li>





</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../09-CADDD/README.html">Module 9: Computer-Aided Drug Discovery and Design</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../09-CADDD/09-1-virtual-screening.html">Virtual Screening</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../10-Machine-Learning/README.html">Module 10: Supervised ML</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../10-Machine-Learning/10-01-supervised-ML.html">Machine Learning Basics</a></li>





</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../appendices/README.html">Appendices</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../appendices/Pandas.html">Introduction to Pandas</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../appendices/MBP1_CellStuctures.html">Molecular Biology Primer for Cheminformatics</a></li>

<li class="toctree-l2"><a class="reference internal" href="../../appendices/MBP2_Genes.html">Introduction to Genes and DNA</a></li>







</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-launch-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Launch interactive content">
    <i class="fas fa-rocket"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://mybinder.org/v2/gh/DivCHED-CCCE/DataChemistry2025OLCC/master?urlpath=lab/tree/docs/content/modules/08-Molecular-Similarity/08-1-MolSimularity.ipynb" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch on Binder"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img alt="Binder logo" src="../../../_static/images/logo_binder.svg">
  </span>
<span class="btn__text-container">Binder</span>
</a>
</li>
      
      
      
      
      <li><a href="https://nanohub.org/tools/datachemolcc/hub/user-redirect/git-pull?repo=https%3A//github.com/DivCHED-CCCE/DataChemistry2025OLCC&urlpath=lab/tree/DataChemistry2025OLCC/docs/content/modules/08-Molecular-Similarity/08-1-MolSimularity.ipynb&branch=master" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch on JupyterHub"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img alt="JupyterHub logo" src="../../../_static/images/logo_jupyterhub.svg">
  </span>
<span class="btn__text-container">JupyterHub</span>
</a>
</li>
      
      
      
      
      <li><a href="https://colab.research.google.com/github/DivCHED-CCCE/DataChemistry2025OLCC/blob/master/docs/content/modules/08-Molecular-Similarity/08-1-MolSimularity.ipynb" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch on Colab"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img alt="Colab logo" src="../../../_static/images/logo_colab.png">
  </span>
<span class="btn__text-container">Colab</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/DivCHED-CCCE/DataChemistry2025OLCC" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/DivCHED-CCCE/DataChemistry2025OLCC/issues/new?title=Issue%20on%20page%20%2Fcontent/modules/08-Molecular-Similarity/08-1-MolSimularity.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../../_sources/content/modules/08-Molecular-Similarity/08-1-MolSimularity.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Fingerprint Generation</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Fingerprint Generation</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#maccs-keys">1: MACCS keys</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#pubchem-fingerprint">2: PubChem Fingerprint</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#circular-fingerprints">1: Circular Fingerprints</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#path-based-fingerprints">2: Path-Based Fingerprints</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#problem-1-maccs-keys">Problem 1: MACCS Keys</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#problem-2">Problem 2</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#problem-3">Problem 3</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#acknowledgments">Acknowledgments</a></li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <center><H1>Generating Molecular Fingerprints</H1></center><div class="alert alert-block alert-info">
<Strong>Objectives</strong>
<ul class="simple">
<li><p>Generate structural keys</p></li>
<li><p>Generate hashed fingerprints</p></li>
</ul>
<p>Many useful documents/papers describe various aspects of molecular similarity, including molecular fingerprints and similarity measures.  Please read these if you need more details.</p>
<ul class="simple">
<li><p>Getting Started with the RDKit in Python<br>
(https://www.rdkit.org/docs/GettingStartedInPython.html#fingerprinting-and-molecular-similarity)</p></li>
<li><p>Fingerprint Generation, GraphSim Toolkit 2.4.2<br>
(https://docs.eyesopen.com/toolkits/python/graphsimtk/fingerprint.html)</p></li>
<li><p>Chemical Fingerprints<br>
(https://docs.chemaxon.com/display/docs/Chemical+Fingerprints)</p></li>
<li><p>Extended-Connectivity Fingerprints<br>
(https://doi.org/10.1021/ci100050t)</p></li>
</ul>
<section id="fingerprint-generation">
<h1>Fingerprint Generation<a class="headerlink" href="#fingerprint-generation" title="Link to this heading">#</a></h1>
<p>Molecular fingerprints are molecular descriptors that encode a molecule’s structure as a bit string. Each bit in the string indicates the presence or absence of a structural feature. This notebook explores generation of two major types of molecular fingerprints:</p>
<ol class="arabic simple">
<li><p>structural keys</p></li>
<li><p>hashed fingerprints</p></li>
</ol>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">rdkit</span><span class="w"> </span><span class="kn">import</span> <span class="n">Chem</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mol</span> <span class="o">=</span> <span class="n">Chem</span><span class="o">.</span><span class="n">MolFromSmiles</span><span class="p">(</span><span class="s1">&#39;CC(C)C1=C(C(=C(N1CC[C@H](C[C@H](CC(=O)O)O)O)C2=CC=C(C=C2)F)C3=CC=CC=C3)C(=O)NC4=CC=CC=C4&#39;</span><span class="p">)</span><span class="c1"># atorvastatin</span>
<span class="n">mol</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-success">
<H3>Structural Keys</H3> <section id="maccs-keys">
<h2>1: MACCS keys<a class="headerlink" href="#maccs-keys" title="Link to this heading">#</a></h2>
<p>The MACCS key is a binary fingerprint (a string of 0’s and 1’s) with a total length of 166 bits.  Each bit position represents the presence (=1) or absence (=0) of a pre-defined structural feature.  The feature definitions for the MACCS keys are available at:<br> https://github.com/rdkit/rdkit/blob/master/rdkit/Chem/MACCSkeys.py</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">rdkit.Chem</span><span class="w"> </span><span class="kn">import</span> <span class="n">MACCSkeys</span>  <span class="c1">#import library for generating MACCSkeys from rdkit</span>
<span class="n">fp</span> <span class="o">=</span> <span class="n">MACCSkeys</span><span class="o">.</span><span class="n">GenMACCSKeys</span><span class="p">(</span><span class="n">mol</span><span class="p">)</span>  <span class="c1"># create an object called fp. It is an instance of the MACCSkeys fingerprint </span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;fp is of type:&#39;</span><span class="p">,(</span><span class="nb">type</span><span class="p">(</span><span class="n">fp</span><span class="p">)))</span>
<span class="nb">print</span><span class="p">()</span>
<span class="c1">#this code prints out each value of the bits in the fingerprint ojbect</span>
<span class="c1">#it loops through each bit and print</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;printing the values of the bitstring as a loop:&#39;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">fp</span><span class="p">)):</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">fp</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">end</span><span class="o">=</span><span class="s1">&#39;&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">()</span>
<span class="c1"># Alternative, easier way to convert it to a bitstring for output.</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;printing the values of the bitstring as ToBitString method in rdkit&#39;</span><span class="p">)</span>
<span class="n">fp</span><span class="o">.</span><span class="n">ToBitString</span><span class="p">()</span>    
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">fp</span><span class="p">))</span>          <span class="c1">#one way to get the number of bits in the fingerprint</span>
<span class="nb">print</span><span class="p">(</span><span class="n">fp</span><span class="o">.</span><span class="n">GetNumBits</span><span class="p">())</span>  <span class="c1"># another way to get the number of bits in the fingerprint</span>
</pre></div>
</div>
</div>
</div>
<p>Note that the MACCS key is <strong>166-bit-long</strong>, but RDKit generates a 167-bit-long fingerprint.  It is because the index of a list/vector in many programming languages (including python) begins at 0.  To use the original numbering of the MACCS keys (1-166) (rather than 0-165), the MACCS keys were implemented to be 167-bit-long, with Bit 0 being always zero. Because Bit 0 is set to OFF for all compounds, it does not affect the evaluation of molecular similarity.</p>
<p>These are some methods that allow you to get some additional information on the MACCS Keys.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">fp</span><span class="o">.</span><span class="n">GetNumBits</span><span class="p">())</span>      <span class="c1">#get total number of bits</span>
<span class="nb">print</span><span class="p">(</span><span class="n">fp</span><span class="o">.</span><span class="n">GetNumOffBits</span><span class="p">())</span>   <span class="c1">#get the total number bits with value 0</span>
<span class="nb">print</span><span class="p">(</span><span class="n">fp</span><span class="o">.</span><span class="n">GetNumOnBits</span><span class="p">())</span>    <span class="c1">#get the total number bits with value 1</span>
<span class="nb">print</span><span class="p">(</span><span class="n">fp</span><span class="o">.</span><span class="n">ToBinary</span><span class="p">())</span>        <span class="c1">#reports the binary representation of the fingerprint</span>
</pre></div>
</div>
</div>
</div>
<p><strong>Run the following code cells.</strong></p>
<ol class="arabic simple">
<li><p>The first cell contains SMILES for a series of molecules and displays their structures</p></li>
<li><p>The second cell displays the MACCS keys for each molecule in the list called smiles.</p></li>
</ol>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># display a series of smiles</span>
<span class="n">smiles</span> <span class="o">=</span> <span class="p">[</span> <span class="s1">&#39;C1=CC=CC=C1&#39;</span><span class="p">,</span> <span class="c1"># Benzene (Kekule)</span>
           <span class="s1">&#39;c1ccccc1&#39;</span><span class="p">,</span>    <span class="c1"># Benzene (&quot;Aromatized&quot; carbons)</span>
          <span class="s1">&#39;C1=CC=NC=C1&#39;</span><span class="p">,</span> <span class="c1"># pyridine   </span>
           <span class="s1">&#39;C1CCCCC1&#39;</span><span class="p">,</span>    <span class="c1"># Cyclohexene</span>
            <span class="s1">&#39;C1CCNCC1&#39;</span><span class="p">]</span>   <span class="c1"># piperidine    </span>

<span class="kn">from</span><span class="w"> </span><span class="nn">rdkit.Chem</span><span class="w"> </span><span class="kn">import</span> <span class="n">Draw</span>
<span class="n">mols</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">smiles</span><span class="p">:</span>
    <span class="n">mols</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Chem</span><span class="o">.</span><span class="n">MolFromSmiles</span><span class="p">(</span><span class="n">x</span><span class="p">))</span>

<span class="c1">#mols = [ Chem.MolFromSmiles(x) for x in smiles ]</span>
<span class="n">Chem</span><span class="o">.</span><span class="n">Draw</span><span class="o">.</span><span class="n">MolsToGridImage</span><span class="p">(</span><span class="n">mols</span><span class="p">,</span> <span class="n">molsPerRow</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">subImgSize</span><span class="o">=</span><span class="p">(</span><span class="mi">100</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span> <span class="n">legends</span><span class="o">=</span><span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">smiles</span><span class="p">]</span> <span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># generate MACCSKeys from smiles</span>
<span class="k">for</span> <span class="n">smile</span> <span class="ow">in</span> <span class="n">smiles</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">smile</span><span class="p">)</span>
    <span class="n">mol</span> <span class="o">=</span> <span class="n">Chem</span><span class="o">.</span><span class="n">MolFromSmiles</span><span class="p">(</span><span class="n">smile</span><span class="p">)</span>
    <span class="n">fp</span> <span class="o">=</span> <span class="n">MACCSkeys</span><span class="o">.</span><span class="n">GenMACCSKeys</span><span class="p">(</span><span class="n">mol</span><span class="p">)</span>
    <span class="n">fp</span> <span class="o">=</span> <span class="n">fp</span><span class="o">.</span><span class="n">ToBitString</span><span class="p">()</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">fp</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">index</span><span class="p">,</span> <span class="n">val</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">fp</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">val</span> <span class="o">==</span> <span class="s1">&#39;1&#39;</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;index is </span><span class="si">%d</span><span class="s2"> and value is </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span><span class="p">(</span><span class="n">index</span><span class="p">,</span> <span class="n">val</span><span class="p">))</span>
    <span class="nb">print</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<strong>Check your understanding</strong>
<p>In general what structural features do benzene and pyridine have in common? <br>
Which MACCS keys do they have in common?</p>
<div class="alert alert-block alert-warning">
The feature definitions for the MACCS keys are available at: https://github.com/rdkit/rdkit/blob/master/rdkit/Chem/MACCSkeys.py <br>
<b>Write the fragment definition of the bits ON that are in common for benzene and pyridine</b> (one is already provided for you as an example).<div class="alert alert-block alert-warning">
<b>Why is index 165 'on' for all of the examples in the smiles list?</b></section>
<section id="pubchem-fingerprint">
<h2>2: PubChem Fingerprint<a class="headerlink" href="#pubchem-fingerprint" title="Link to this heading">#</a></h2>
<p>The PubChem Fingerprint is a 881-bit-long binary fingerprint (ftp://ftp.ncbi.nlm.nih.gov/pubchem/specifications/pubchem_fingerprints.pdf).  Similar to the MACCS keys, it uses a pre-defined fragment dictionary.  The PubChem fingerprint for each compound in PubChem can be downloaded from PubChem.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># the following code cell generates a PUG REST request to obtain the PubChem Fingerprint</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">requests</span>
<span class="n">prolog</span> <span class="o">=</span> <span class="s2">&quot;https://pubchem.ncbi.nlm.nih.gov/rest/pug&quot;</span>
<span class="n">cid</span> <span class="o">=</span><span class="s2">&quot;2244&quot;</span>  <span class="c1"># CID for aspirin </span>
<span class="n">url</span> <span class="o">=</span> <span class="n">prolog</span> <span class="o">+</span> <span class="s2">&quot;/compound/cid/&quot;</span> <span class="o">+</span> <span class="n">cid</span> <span class="o">+</span> <span class="s2">&quot;/property/Fingerprint2D/TXT&quot;</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">url</span><span class="p">)</span>
<span class="n">pcfp_base64</span> <span class="o">=</span> <span class="n">res</span><span class="o">.</span><span class="n">text</span>  <span class="c1">#pcfp = PubChemFingerPrint</span>
<span class="nb">print</span><span class="p">(</span><span class="n">pcfp_base64</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Notice that the above output is not a binary bitstring. Pubchem Fingerprints are provided as Base64 encoded strings. The PubChem fingerprint is 881 bits representing structural features. Instead of storing these bits directly, PubChem encodes them as a new Base64 string. This provides for safer and easier transmittion through application programming interfaces (API) like PUG REST and databases can handle text better than the raw binary ones and zeros. When retrieved by the PUG REST API, the fingerprint appears as a 157 character string. Because they are base64-encoded, they should be decoded into binary bitstrings or bitvectors.</p>
<p>Details about how to decode base64-encoded PubChem fingerprints is described on the last page of the PubChem Fingerprint specification (https://ftp.ncbi.nlm.nih.gov/pubchem/specifications/pubchem_fingerprints.pdf).  Below is a user-defined function that decodes a PubChem fingerprint into a bit string.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">base64</span><span class="w"> </span><span class="kn">import</span> <span class="n">b64decode</span>

<span class="k">def</span><span class="w"> </span><span class="nf">PCFP_BitString</span><span class="p">(</span><span class="n">pcfp_base64</span><span class="p">)</span> <span class="p">:</span>

    <span class="n">pcfp_bitstring</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="p">[</span><span class="s2">&quot;</span><span class="si">{:08b}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">b64decode</span><span class="p">(</span> <span class="n">pcfp_base64</span> <span class="p">)]</span> <span class="p">)[</span><span class="mi">32</span><span class="p">:</span><span class="mi">913</span><span class="p">]</span>
    <span class="k">return</span> <span class="n">pcfp_bitstring</span>
    
</pre></div>
</div>
</div>
</div>
<p>Using the user defined function we created above, we can convert the base output to binary.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pcfp_bitstring</span> <span class="o">=</span> <span class="n">PCFP_BitString</span><span class="p">(</span><span class="n">pcfp_base64</span><span class="p">)</span>  <span class="c1"># use the user defined function to convert the PubChem FingerPrint to a bitstring</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">pcfp_bitstring</span><span class="p">))</span>
<span class="nb">print</span><span class="p">(</span><span class="n">pcfp_bitstring</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>The generated bitstring can be converted to a bitvector that can be used for molecular similarity computation in RDKit (to be discussed in part 2).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">rdkit</span><span class="w"> </span><span class="kn">import</span> <span class="n">DataStructs</span>
<span class="n">bitvect</span> <span class="o">=</span> <span class="n">DataStructs</span><span class="o">.</span><span class="n">CreateFromBitString</span><span class="p">(</span><span class="n">PCFP_BitString</span><span class="p">(</span><span class="n">pcfp_base64</span><span class="p">))</span>
<span class="nb">type</span><span class="p">(</span><span class="n">bitvect</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-success">
<H3>Hashed Fingerprints</H3> </section>
<section id="circular-fingerprints">
<h2>1: Circular Fingerprints<a class="headerlink" href="#circular-fingerprints" title="Link to this heading">#</a></h2>
<p>MACCS Keys and PubChem Fingerprints are examples of structural keys which use a fixed length bit vector (166 and 881, respectively). Each bit in the vector corresponds to the presence (1) or absence (0) of a predefined chemical feature (atoms, bonds) or substructure ( aromatic rings, carbonyl). In contrast, extended connectivity fingerprints are generated algorithmically by exploring the neighborhood or each atom up to a given radius. The environment is then encoded into hashed identifiers. Structural keys rely on a fixed dictionary of features. Extnded connectivity fingerprints are more flexible and capture local atomic environments without the need for predefined substructures.</p>
<p>Circular fingerprints are hashed fingerprints.  They are generated by exhaustively enumerating “circular” fragments (containing all atoms within a given radius from each heavy atom of the molecule) and then hashing these fragments into a fixed-length bitstring.  (Here, the “radius” from an atom is measured by the number of bonds that separates two atoms).</p>
<p>Examples of circular fingerprints are the extended-connectivity fingerprint (ECFPs) and their variant called FCFPs (Functional-Class Fingerprints), originally described in a paper by Rogers and Hahn (https://doi.org/10.1021/ci100050t).  The RDKit implementation of these fingerprints are called “Morgan Fingerprints” (https://www.rdkit.org/docs/GettingStartedInPython.html#morgan-fingerprints-circular-fingerprints).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fluoxetine</span> <span class="o">=</span> <span class="n">Chem</span><span class="o">.</span><span class="n">MolFromSmiles</span><span class="p">(</span><span class="s1">&#39;CNCCC(C1=CC=CC=C1)OC2=CC=C(C=C2)C(F)(F)F&#39;</span><span class="p">)</span>
<span class="n">fluoxetine</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">rdkit.Chem</span><span class="w"> </span><span class="kn">import</span> <span class="n">rdFingerprintGenerator</span>
<span class="n">mfpgen</span> <span class="o">=</span> <span class="n">rdFingerprintGenerator</span><span class="o">.</span><span class="n">GetMorganGenerator</span><span class="p">(</span><span class="n">radius</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span><span class="n">fpSize</span><span class="o">=</span><span class="mi">2048</span><span class="p">)</span>
<span class="n">fp2</span> <span class="o">=</span> <span class="n">mfpgen</span><span class="o">.</span><span class="n">GetFingerprint</span><span class="p">(</span><span class="n">fluoxetine</span><span class="p">)</span>
<span class="n">bitstring</span><span class="o">=</span> <span class="n">fp2</span><span class="o">.</span><span class="n">ToBitString</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="n">bitstring</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;The fingerprint length is&quot;</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">bitstring</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<p>When comparing the RDK’s Morgan fingerprints with the ECFP/FCFP fingerprints, it is important to remember that the name of ECFP/FCFP fingerprints are suffixed with the <strong>diameter</strong> of the atom environments considered, while the Morgan Fingerprints take a <strong>radius</strong> parameter (e.g., the second argument “2” of GetMorganFingerprintAsBitVect() in the above code cell).  The Morgan fingerprint generated above (with a radius of 2) is comparable to the ECFP4 fingerprint (with a diameter of 4).</p>
<p>MACCS Keys and PubChem Fingerprints are examples of structural keys which use a fixed length bit vector (166 and 881, respectively). Each bit in the vector corresponds to the presence (1) or absence (0) of a predefined chemical feature (atoms, bonds) or substructure ( aromatic rings, carbonyl). In contrast, extended connectivity fingerprints are generated algorithmically by exploring the neighborhood or each atom up to a given radius. The environment is then encoded into hashed identifiers. Structural keys rely on a fixed dictionary of features. Extnded connectivity fingerprints are more flexible and capture local atomic environments without the need for predefined substructures.</p>
<p><strong>To get a better idea of how these fingerprints are generated, over this next section you will explore 1-bromobutane and 1-chlorobutane</strong></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">rdkit.Chem</span><span class="w"> </span><span class="kn">import</span> <span class="n">Draw</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">rdkit.Chem.Draw</span><span class="w"> </span><span class="kn">import</span> <span class="n">IPythonConsole</span>
<span class="n">IPythonConsole</span><span class="o">.</span><span class="n">ipython_useSVG</span> <span class="o">=</span> <span class="kc">True</span>
<span class="n">IPythonConsole</span><span class="o">.</span><span class="n">drawOptions</span><span class="o">.</span><span class="n">addAtomIndices</span> <span class="o">=</span> <span class="kc">True</span>   <span class="c1">#this will add numbers to the image to help identify carbons later</span>
<span class="n">IPythonConsole</span><span class="o">.</span><span class="n">drawOptions</span><span class="o">.</span><span class="n">addStereoAnnotation</span> <span class="o">=</span> <span class="kc">False</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mol</span> <span class="o">=</span> <span class="n">Chem</span><span class="o">.</span><span class="n">MolFromSmiles</span><span class="p">(</span><span class="s2">&quot;BrCCCC&quot;</span><span class="p">)</span>  <span class="c1"># 1-bromobutane</span>
<span class="n">mol</span>
</pre></div>
</div>
</div>
</div>
<p>Let’s generate its rdkit Morgan Fingerprint (radius = 2). This would be comparable to the ECFP4 fingerprint.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">rdkit</span><span class="w"> </span><span class="kn">import</span> <span class="n">Chem</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">rdkit.Chem</span><span class="w"> </span><span class="kn">import</span> <span class="n">rdFingerprintGenerator</span>

<span class="c1"># Step 1: Create the molecule from SMILES</span>
<span class="n">mol</span> <span class="o">=</span> <span class="n">Chem</span><span class="o">.</span><span class="n">MolFromSmiles</span><span class="p">(</span><span class="s1">&#39;BrCCCC&#39;</span><span class="p">)</span>  <span class="c1"># 1-bromobutane</span>
<span class="c1">#mol = Chem.MolFromSmiles(&#39;CNC[C@H](O)c1ccc(O)c(O)c1&#39;) #epinephrine</span>

<span class="c1"># Step 2: Initialize the Morgan fingerprint generator. We are setting the radius to 2 and total bit size to 1024</span>
<span class="n">morgan_gen</span> <span class="o">=</span> <span class="n">rdFingerprintGenerator</span><span class="o">.</span><span class="n">GetMorganGenerator</span><span class="p">(</span><span class="n">radius</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">fpSize</span><span class="o">=</span><span class="mi">1024</span><span class="p">)</span>

<span class="c1"># Step 3: Prepare the AdditionalOutput object to capture bit information</span>
<span class="n">additional_output</span> <span class="o">=</span> <span class="n">rdFingerprintGenerator</span><span class="o">.</span><span class="n">AdditionalOutput</span><span class="p">()</span>
<span class="n">additional_output</span><span class="o">.</span><span class="n">AllocateBitInfoMap</span><span class="p">()</span>

<span class="c1"># Step 4: Generate the fingerprint with additional output</span>
<span class="n">fp1</span> <span class="o">=</span> <span class="n">morgan_gen</span><span class="o">.</span><span class="n">GetFingerprint</span><span class="p">(</span><span class="n">mol</span><span class="p">,</span> <span class="n">additionalOutput</span><span class="o">=</span><span class="n">additional_output</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;The Morgan Fingerprint radius 2 fingerprint (ECPF4) for the molecule is:&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">fp1</span><span class="o">.</span><span class="n">ToBitString</span><span class="p">())</span>
<span class="nb">print</span><span class="p">()</span>

<span class="c1"># Step 5: Retrieve and display the bit information</span>

<span class="n">bit_info</span> <span class="o">=</span> <span class="n">additional_output</span><span class="o">.</span><span class="n">GetBitInfoMap</span><span class="p">()</span>
<span class="k">for</span> <span class="n">bit_id</span><span class="p">,</span> <span class="n">atom_radius_list</span> <span class="ow">in</span> <span class="n">bit_info</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Bit </span><span class="si">{</span><span class="n">bit_id</span><span class="si">}</span><span class="s2"> is set by:&quot;</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">atom_idx</span><span class="p">,</span> <span class="n">radius</span> <span class="ow">in</span> <span class="n">atom_radius_list</span><span class="p">:</span>
        <span class="n">atom_symbol</span> <span class="o">=</span> <span class="n">mol</span><span class="o">.</span><span class="n">GetAtomWithIdx</span><span class="p">(</span><span class="n">atom_idx</span><span class="p">)</span><span class="o">.</span><span class="n">GetSymbol</span><span class="p">()</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;  - Atom index </span><span class="si">{</span><span class="n">atom_idx</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="n">atom_symbol</span><span class="si">}</span><span class="s2">), Radius </span><span class="si">{</span><span class="n">radius</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p><strong>We can also display an image of the fragment that caused the bit to be equal to 1.</strong><br></p>
<p>Some notes about rendering:</p>
<ul class="simple">
<li><p>The molecule fragment is drawn with the atoms in the same positions as in the original molecule.</p></li>
<li><p>The central atom is highlighted in blue.</p></li>
<li><p>Aromatic atoms are highlighted in yellow</p></li>
<li><p>Aliphatic ring atoms are highlighted in dark gray</p></li>
<li><p>Atoms/bonds that are drawn in light gray indicate pieces of the structure that influence the atoms’ connectivity invariants but that are not directly part of the fingerprint.</p></li>
</ul>
<p>As an example, we can draw the fragment for bit 80 below (and change to different values based on the above bitlist.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">IPythonConsole</span><span class="o">.</span><span class="n">drawOptions</span><span class="o">.</span><span class="n">addAtomIndices</span> <span class="o">=</span> <span class="kc">False</span> 
<span class="n">mfp2_svg</span> <span class="o">=</span> <span class="n">Draw</span><span class="o">.</span><span class="n">DrawMorganBit</span><span class="p">(</span><span class="n">mol</span><span class="p">,</span> <span class="mi">80</span><span class="p">,</span> <span class="n">bit_info</span><span class="p">,</span> <span class="n">useSVG</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">mfp2_svg</span>
</pre></div>
</div>
</div>
</div>
<p>While the above code can display 1 fingerprint fragment, it is more useful to display all fragments simultaneously:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create a list of tuples for visualization</span>
<span class="n">tpls</span> <span class="o">=</span> <span class="p">[(</span><span class="n">mol</span><span class="p">,</span> <span class="n">bit_id</span><span class="p">,</span> <span class="n">bit_info</span><span class="p">)</span> <span class="k">for</span> <span class="n">bit_id</span> <span class="ow">in</span> <span class="n">fp1</span><span class="o">.</span><span class="n">GetOnBits</span><span class="p">()]</span>
<span class="c1"># Generate legends for each bit</span>
<span class="n">legends</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">bit_id</span><span class="p">)</span> <span class="k">for</span> <span class="n">bit_id</span> <span class="ow">in</span> <span class="n">fp1</span><span class="o">.</span><span class="n">GetOnBits</span><span class="p">()]</span>
<span class="c1"># Visualize the bits</span>
<span class="n">Draw</span><span class="o">.</span><span class="n">DrawMorganBits</span><span class="p">(</span><span class="n">tpls</span><span class="p">,</span> <span class="n">molsPerRow</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">legends</span><span class="o">=</span><span class="n">legends</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<strong>Check your understanding</strong>
<ul class="simple">
<li><p>Which bit above represents the bromine with a radius of 0?</p></li>
<li><p>Why do three carbons in 1-bromobutane result in an “on” bit for fragment 80?</p></li>
<li><p>If you changed 1-bromobutane to 1-chlorobutane, which fragments above (33, 80, 251, 294, 375, 495, 591, 640, 728, 794, 887) would you still expect to have a value of 1? (write this prediction down in the next cell for later)</p></li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your prediction for which values would you still expect to have a value of 1 if you changed the molecule from 1-bromobutane to 1-chlorobutane</span>
</pre></div>
</div>
</div>
</div>
<p>Let’s compare the 1-bromobutane to 1-chlorobutane to determine if they have any fragments in common.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">IPythonConsole</span><span class="o">.</span><span class="n">drawOptions</span><span class="o">.</span><span class="n">addAtomIndices</span> <span class="o">=</span> <span class="kc">True</span> 
<span class="n">mol</span> <span class="o">=</span> <span class="n">Chem</span><span class="o">.</span><span class="n">MolFromSmiles</span><span class="p">(</span><span class="s1">&#39;ClCCCC&#39;</span><span class="p">)</span>  <span class="c1"># 1-chlorobutane</span>
<span class="n">mol</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Step 1: Create the molecule from SMILES</span>
<span class="n">mol</span> <span class="o">=</span> <span class="n">Chem</span><span class="o">.</span><span class="n">MolFromSmiles</span><span class="p">(</span><span class="s1">&#39;ClCCCC&#39;</span><span class="p">)</span>  <span class="c1"># 1-chlorobutane</span>

<span class="c1"># Step 2: Initialize the Morgan fingerprint generator. We are setting the radius to 2 and total bit size to 1024</span>
<span class="n">morgan_gen</span> <span class="o">=</span> <span class="n">rdFingerprintGenerator</span><span class="o">.</span><span class="n">GetMorganGenerator</span><span class="p">(</span><span class="n">radius</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">fpSize</span><span class="o">=</span><span class="mi">1024</span><span class="p">)</span>

<span class="c1"># Step 3: Prepare the AdditionalOutput object to capture bit information</span>
<span class="n">additional_output</span> <span class="o">=</span> <span class="n">rdFingerprintGenerator</span><span class="o">.</span><span class="n">AdditionalOutput</span><span class="p">()</span>
<span class="n">additional_output</span><span class="o">.</span><span class="n">AllocateBitInfoMap</span><span class="p">()</span>

<span class="c1"># Step 4: Generate the fingerprint with additional output</span>
<span class="n">fp2</span> <span class="o">=</span> <span class="n">morgan_gen</span><span class="o">.</span><span class="n">GetFingerprint</span><span class="p">(</span><span class="n">mol</span><span class="p">,</span> <span class="n">additionalOutput</span><span class="o">=</span><span class="n">additional_output</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;The Morgan radius =2 fingerprint (ECFP4)for the molecule is:&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">fp2</span><span class="o">.</span><span class="n">ToBitString</span><span class="p">())</span>
<span class="nb">print</span><span class="p">()</span>

<span class="c1"># Step 5: Retrieve and display the bit information</span>

<span class="n">bit_info</span> <span class="o">=</span> <span class="n">additional_output</span><span class="o">.</span><span class="n">GetBitInfoMap</span><span class="p">()</span>
<span class="k">for</span> <span class="n">bit_id</span><span class="p">,</span> <span class="n">atom_radius_list</span> <span class="ow">in</span> <span class="n">bit_info</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Bit </span><span class="si">{</span><span class="n">bit_id</span><span class="si">}</span><span class="s2"> is set by:&quot;</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">atom_idx</span><span class="p">,</span> <span class="n">radius</span> <span class="ow">in</span> <span class="n">atom_radius_list</span><span class="p">:</span>
        <span class="n">atom_symbol</span> <span class="o">=</span> <span class="n">mol</span><span class="o">.</span><span class="n">GetAtomWithIdx</span><span class="p">(</span><span class="n">atom_idx</span><span class="p">)</span><span class="o">.</span><span class="n">GetSymbol</span><span class="p">()</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;  - Atom index </span><span class="si">{</span><span class="n">atom_idx</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="n">atom_symbol</span><span class="si">}</span><span class="s2">), Radius </span><span class="si">{</span><span class="n">radius</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">IPythonConsole</span><span class="o">.</span><span class="n">drawOptions</span><span class="o">.</span><span class="n">addAtomIndices</span> <span class="o">=</span> <span class="kc">False</span> 
<span class="c1"># Create a list of tuples for visualization</span>
<span class="n">tpls</span> <span class="o">=</span> <span class="p">[(</span><span class="n">mol</span><span class="p">,</span> <span class="n">bit_id</span><span class="p">,</span> <span class="n">bit_info</span><span class="p">)</span> <span class="k">for</span> <span class="n">bit_id</span> <span class="ow">in</span> <span class="n">fp2</span><span class="o">.</span><span class="n">GetOnBits</span><span class="p">()]</span>
<span class="c1"># Generate legends for each bit</span>
<span class="n">legends</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">bit_id</span><span class="p">)</span> <span class="k">for</span> <span class="n">bit_id</span> <span class="ow">in</span> <span class="n">fp2</span><span class="o">.</span><span class="n">GetOnBits</span><span class="p">()]</span>
<span class="c1"># Visualize the bits</span>
<span class="n">Draw</span><span class="o">.</span><span class="n">DrawMorganBits</span><span class="p">(</span><span class="n">tpls</span><span class="p">,</span> <span class="n">molsPerRow</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">legends</span><span class="o">=</span><span class="n">legends</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Now let’s identify the bits it common of 1-bromobutane and 1-chlorobutane!</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">common_bits</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">fp1</span><span class="o">.</span><span class="n">GetOnBits</span><span class="p">())</span> <span class="o">&amp;</span> <span class="nb">set</span><span class="p">(</span><span class="n">fp2</span><span class="o">.</span><span class="n">GetOnBits</span><span class="p">())</span> 
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Common bits: </span><span class="si">{</span><span class="nb">sorted</span><span class="p">(</span><span class="n">common_bits</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="alert alert-block alert-warning">
<strong>Check your understanding</strong>
<p>Did your prediction from the previous check your understanding hold true? If not, review the data to make sure you understand which fragments are on in both.</p>
</section>
<section id="path-based-fingerprints">
<h2>2: Path-Based Fingerprints<a class="headerlink" href="#path-based-fingerprints" title="Link to this heading">#</a></h2>
<p>Path-based fingerprints are also hashed fingerprints.  They are generated by enumerating linear fragments of a given length and hashing them into a fixed-length bitstring.  An example is the RDKit’s topological fingeprint.  As described in the RDK documentation (https://www.rdkit.org/docs/GettingStartedInPython.html#topological-fingerprints), while this fingerprint can be generated using FingerprintMols.FingerprintMol(), it is recommended to use rdmolops.RDKFingerprint() to generate the fingerprint using non-default parameter values.</p>
<p>The <a href ="https://www.rdkit.org/docs/source/rdkit.Chem.rdmolops.html#:~:text=rdkit.Chem.rdmolops.RDKFingerprint((Mol)mol">RDKFingerprint(mol) method</a> has a number of arguments that can be added:</p>
<ul class="simple">
<li><p>mol: the molecule to use</p></li>
<li><p>fpSize: (optional) number of bits in the fingerprint Defaults to 2048.</p></li>
<li><p>minPath: (optional) minimum number of bonds to include in the subgraphs Defaults to 1.</p></li>
<li><p>maxPath: (optional) maximum number of bonds to include in the subgraphs Defaults to 7.</p></li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># path based fingerprints</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">rdkit.Chem</span><span class="w"> </span><span class="kn">import</span> <span class="n">rdmolops</span>
<span class="n">mol</span> <span class="o">=</span> <span class="n">Chem</span><span class="o">.</span><span class="n">MolFromSmiles</span><span class="p">(</span><span class="s2">&quot;CCOC(=O)N1CCC(=C2C3=C(CCC4=C2N=CC=C4)C=C(C=C3)Cl)CC1&quot;</span><span class="p">)</span>  <span class="c1"># loratadine</span>
<span class="n">fp</span> <span class="o">=</span> <span class="n">rdmolops</span><span class="o">.</span><span class="n">RDKFingerprint</span><span class="p">(</span><span class="n">mol</span><span class="p">,</span> <span class="n">fpSize</span><span class="o">=</span><span class="mi">2048</span><span class="p">,</span> <span class="n">minPath</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">maxPath</span><span class="o">=</span><span class="mi">7</span><span class="p">)</span><span class="o">.</span><span class="n">ToBitString</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="n">fp</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
 <div class="alert alert-block alert-success">
      <H1><center>Homework</center></H1></section>
</section>
<section id="problem-1-maccs-keys">
<h1>Problem 1: MACCS Keys<a class="headerlink" href="#problem-1-maccs-keys" title="Link to this heading">#</a></h1>
<p>For the list of terpene SMILES in the next code cell:</p>
<ol class="arabic simple">
<li><p>Display the structures</p></li>
<li><p>Generate the MACCS Keys for each molecule</p></li>
<li><p>Calculate number of “on” bits and “off” bits</p></li>
</ol>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">terpenes</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;CC(=CCCC(C)(C=C)O)C&#39;</span><span class="p">,</span>                       <span class="c1">#linalool</span>
          <span class="s1">&#39;CC(=CCC/C(=C/CO)/C)C&#39;</span><span class="p">,</span>                        <span class="c1">#geraniol</span>
          <span class="s1">&#39;CC1=CCC(CC1)C(=C)C&#39;</span><span class="p">,</span>                          <span class="c1">#limonene</span>
          <span class="s1">&#39;CC1=CCC(=C(C)C)CC1&#39;</span><span class="p">,</span>                          <span class="c1">#terpinolene</span>
          <span class="s1">&#39;CC1=CCC2CC1C2(C)C&#39;</span><span class="p">]</span>                           <span class="c1">#Alpha-pinene</span>

<span class="c1"># write your code here to display the molecules in Problem 1</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># write your code here to generate MACCS Keys and off/on bits in Problem 1</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="problem-2">
<h1>Problem 2<a class="headerlink" href="#problem-2" title="Link to this heading">#</a></h1>
<p>For the list of PubChem compound ID numbers in the next code cell:</p>
<ol class="arabic simple">
<li><p>Use the use the PUG-REST API to obtain their SMILES and PubChem Fingerprints and store in lists</p></li>
<li><p>convert the PubChem Fingerprints to bitstrings and display</p></li>
<li><p>display their structures</p></li>
</ol>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">CIDS</span> <span class="o">=</span> <span class="p">[</span>     <span class="mi">4980</span><span class="p">,</span>   <span class="c1"># psilocin             </span>
             <span class="mi">1615</span><span class="p">,</span>   <span class="c1"># methylenedioxymethamphetamine</span>
            <span class="mi">10257</span><span class="p">,</span>   <span class="c1"># bufotenin</span>
           <span class="mi">360252</span><span class="p">,</span>   <span class="c1"># 5-bromo-DMT  </span>
             <span class="mi">4076</span><span class="p">,</span>   <span class="c1"># mescaline </span>
            <span class="mi">98527</span><span class="p">,</span>   <span class="c1"># 2C-B</span>
            <span class="mi">5761</span> <span class="p">]</span>   <span class="c1"># LSD </span>

<span class="c1"># write your code here to obtain SMILES and PubChem Fingerprints </span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># write your code here to convert fingerprints from Base64 to bitstring</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># write your code here to display structures</span>
</pre></div>
</div>
</div>
</div>
<section id="problem-3">
<h2>Problem 3<a class="headerlink" href="#problem-3" title="Link to this heading">#</a></h2>
<p>For the molecules below, generate the 512-bit-long Morgan Fingeprint comparable to the <strong>FCFP6</strong> fingerprint.</p>
<ul class="simple">
<li><p>Use the PUG REST API to search for the compounds by name and get their SMILES strings.</p></li>
<li><p>Generate the molecular fingerprints from the SMILES strings.</p></li>
<li><p>Print the generated fingerprints.</p></li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">synonyms</span> <span class="o">=</span> <span class="p">[</span> <span class="s1">&#39;diphenhydramine&#39;</span><span class="p">,</span> <span class="s1">&#39;cetirizine&#39;</span><span class="p">,</span> <span class="s1">&#39;fexofenadine&#39;</span><span class="p">,</span> <span class="s1">&#39;loratadine&#39;</span> <span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write your code in this cell</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="acknowledgments">
<h1>Acknowledgments<a class="headerlink" href="#acknowledgments" title="Link to this heading">#</a></h1>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "datachem"
        },
        kernelOptions: {
            name: "datachem",
            path: "./content/modules/08-Molecular-Similarity"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'datachem'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="README.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Module 8: Molecular Similarity</p>
      </div>
    </a>
    <a class="right-next"
       href="08-2-MolSimularity.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">What is a bit vector?</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Fingerprint Generation</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#maccs-keys">1: MACCS keys</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#pubchem-fingerprint">2: PubChem Fingerprint</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#circular-fingerprints">1: Circular Fingerprints</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#path-based-fingerprints">2: Path-Based Fingerprints</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#problem-1-maccs-keys">Problem 1: MACCS Keys</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#problem-2">Problem 2</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#problem-3">Problem 3</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#acknowledgments">Acknowledgments</a></li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Robert Belford
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>